---
title: "DSEQ2V2"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

path="~/Users/gallifrey/Desktop/dnFGFR/"
setwd("/Users/gallifrey/Desktop/dnFGFR/")
```

#Import Libraries
```{r}
knitr::opts_chunk$set(echo = TRUE)
require(data.table)
require(DESeq2)
require(RColorBrewer)
require(EDASeq)
require(cqn)
require(AnnotationDbi)
require(org.Mm.eg.db)
require(pheatmap)
require(genefilter)
require(WGCNA)
require(GO.db)
require(knitr)
require(GenomicFeatures)
require(BiocParallel)
require(limma)
require(ggplot2)
ALLOW_WGCNA_THREADS=8
knitr::opts_knit$set(root.dir ="/Users/gallifrey/Desktop/dnFGFR/")
```

#Load counts and sample info
```{r}
counts = read.table("allSamplesFixed.txt", header=TRUE, row.names=1, sep="\t")
coldata = read.table("coldataAllSamplesFixed.txt", header=TRUE, sep="\t")
```

#remove length from counts table
```{r}
counts['Length'] = NULL
```

#sort coldata so it matches counts table columns
```{r}
#coldata_sort <-coldata[order(coldata$sample),]
```

#make deseq2 object
```{r}
dds = DESeqDataSetFromMatrix(countData = counts, colData = coldata, design= ~housing + genotype:time)
```

#transform counts for PCA plots
```{r}
rld <- rlog(dds, blind=FALSE)
vst <- vst(dds, blind=FALSE)
ntd <- normTransform(dds)
```

#plot SD of transformed data
```{r}
library("vsn")
pdf("/Users/gallifrey/Desktop/dnFGFR/sdplot_shiftedLogtransformed.pdf")
meanSdPlot(assay(ntd))
dev.off()

pdf("/Users/gallifrey/Desktop/dnFGFR/sdplot_regLogtransformed.pdf")
meanSdPlot(assay(rld))
dev.off()

pdf("/Users/gallifrey/Desktop/dnFGFR/sdplot_varstabilizedtransformed.pdf")
meanSdPlot(assay(vst))
dev.off()
```

#check to see what is the dominant driver of gene expression differences
```{r}
pdf("allsamples_genotype_PCA.pdf")
sampleDists <- dist(t(assay(vst)))
plotPCA(rld, intgroup=c("genotype"))
dev.off()

pdf("allsamples_treatment_PCA.pdf")
sampleDists <- dist(t(assay(vst)))
plotPCA(rld, intgroup=c("housing"))
dev.off()

pdf("allsamples_day_PCA.pdf")
sampleDists <- dist(t(assay(vst)))
plotPCA(rld, intgroup=c("time"))
dev.off()
```

#check to see what is the dominant driver of gene expression differences, another PCA example
```{r}
#plotPCA(vst, intgroup=c("day"),ntop=100)
pcaData <- plotPCA(vst, intgroup=c("genotype", "housing"), returnData=TRUE)
percentVar <- round(100 * attr(pcaData, "percentVar"))
ggplot(pcaData, aes(PC1, PC2, color=housing, shape=genotype)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance")) + 
  coord_fixed()
```


# PCA FIXED All samples, effect of genotype controlling for housing and time
```{r}
#Designate Files
setwd("/Users/gallifrey/Desktop/dnFGFR/interaction/")

counts1 = read.table("allSamplesFixed.txt", header=TRUE, row.names=1, sep="\t")
coldata = read.table("coldataAllSamplesFixed.txt", header=TRUE, sep="\t")

#remove length column
counts1['Length'] = NULL

#set dds object, define design
dds = DESeqDataSetFromMatrix(countData = counts1, colData = coldata, design = ~ housing + time:genotype)

#remove low count genes (anything less than 20 reads)
keep <- rowSums(counts(dds)) >= 20
dds <- dds[keep,]

#set comparison level, 
dds$housing <- relevel(dds$housing, ref = "same")
dds$genotype <- relevel(dds$genotype, ref = "ctrl")

#perform normalization, dispersion estimate, and statistical tests
dds = DESeq(dds)

#plot dispersions
pdf("all_fixed_dispersionPlot.pdf")
plotDispEsts(dds )
dev.off()

#write out normalized counts
normed = counts(dds, normalized=TRUE)
write.csv(normed, file="all_fixed_genotype_normCounts.csv")

#write out statistical results
res <- results(dds)
write.csv(as.data.frame(res), file="all_fixed_genotype_results.csv")

#plot fold changes with 2 fold blue line added, alpha is FDR cutoff, change to whatever cutoff you want
pdf("all_fixed_genotype_MAPlot.pdf")
DESeq2::plotMA(res, alpha = 0.1, ylim=c(-3,3), cex=.4)
abline(h=c(-1,1), col="dodgerblue", lwd=2)
dev.off()

```

# PCA FIXED All samples, effect of housing controlling for genotype and time
```{r}
#Designate Files
setwd("/Users/gallifrey/Desktop/dnFGFR/allsamplesfixed/deseq2/genotype/")

counts1 = read.table("allSamplesFixed.txt", header=TRUE, row.names=1, sep="\t")
coldata = read.table("coldataAllSamplesFixed.txt", header=TRUE, sep="\t")

#remove length column
counts1['Length'] = NULL

#set dds object, define design
dds = DESeqDataSetFromMatrix(countData = counts1, colData = coldata, design = ~ housing + time + genotype)

#remove low count genes (anything less than 20 reads)
keep <- rowSums(counts(dds)) >= 20
dds <- dds[keep,]

#set comparison level, 
dds$housing <- relevel(dds$housing, ref = "same")
dds$genotype <- relevel(dds$genotype, ref = "ctrl")

#perform normalization, dispersion estimate, and statistical tests
dds = DESeq(dds)

#plot dispersions
pdf("all_fixed_dispersionPlot.pdf")
plotDispEsts(dds )
dev.off()

#write out normalized counts
normed = counts(dds, normalized=TRUE)
write.csv(normed, file="all_fixed_genotype_normCounts.csv")

#write out statistical results
res <- results(dds)
write.csv(as.data.frame(res), file="all_fixed_genotype_results.csv")

#plot fold changes with 2 fold blue line added, alpha is FDR cutoff, change to whatever cutoff you want
pdf("all_fixed_genotype_MAPlot.pdf")
DESeq2::plotMA(res, alpha = 0.1, ylim=c(-3,3), cex=.4)
abline(h=c(-1,1), col="dodgerblue", lwd=2)
dev.off()

```



# PCA FIXED All samples, effect of interaction term housing*time
```{r}
#Designate Files
setwd("/Users/gallifrey/Desktop/dnFGFR/allsamplesfixed/deseq2/interaction/")

counts1 = read.table("allSamplesFixed.txt", header=TRUE, row.names=1, sep="\t")
coldata = read.table("coldataAllSamplesFixed.txt", header=TRUE, sep="\t")

#remove length column
counts1['Length'] = NULL

#set dds object, define design
dds = DESeqDataSetFromMatrix(countData = counts1, colData = coldata, design = ~ housing + time + genotype + housing:genotype)

#remove low count genes (anything less than 20 reads)
keep <- rowSums(counts(dds)) >= 20
dds <- dds[keep,]

#set comparison level, 
dds$housing <- relevel(dds$housing, ref = "same")
dds$genotype <- relevel(dds$genotype, ref = "ctrl")

#perform normalization, dispersion estimate, and statistical tests
dds = DESeq(dds)

#plot dispersions
pdf("all_fixed_dispersionPlot.pdf")
plotDispEsts(dds )
dev.off()

#write out normalized counts
normed = counts(dds, normalized=TRUE)
write.csv(normed, file="all_fixed_interaction_normCounts.csv")

#write out statistical results
res <- results(dds)
write.csv(as.data.frame(res), file="all_fixed_interaction_results.csv")

#plot fold changes with 2 fold blue line added, alpha is FDR cutoff, change to whatever cutoff you want
pdf("all_fixed_interaction_MAPlot.pdf")
DESeq2::plotMA(res, alpha = 0.1, ylim=c(-3,3), cex=.4)
abline(h=c(-1,1), col="dodgerblue", lwd=2)
dev.off()

```




# PN140 dnFGFR effect of housing
```{r}
setwd("/Users/gallifrey/Desktop/dnFGFR/PN140 dnFGFR/")
#Designate Files
counts1 = read.table("PN140dnFGFRcounts.txt", header=TRUE, row.names=1, sep="\t")
coldata = read.table("PN140dnFGFRcoldata.txt", header=TRUE, row.names=1, sep="\t")

#remove length column
counts1['Length'] = NULL

#set dds object, define design
dds = DESeqDataSetFromMatrix(countData = counts1, colData = coldata, design = ~housing)

#remove low count genes (anything less than 20 reads)
keep <- rowSums(counts(dds)) >= 20
dds <- dds[keep,]

#set comparison level, 
dds$housing <- relevel(dds$housing, ref = "same")

#perform normalization, dispersion estimate, and statistical tests
dds = DESeq(dds)

#plot dispersions
pdf("PN140dnFGFRdispersionPlot.pdf")
plotDispEsts(dds )
dev.off()

#write out normalized counts
normed = counts(dds, normalized=TRUE)
write.csv(normed, file="PN140dnFGFRNormCounts.csv")

#write out statistical results
res <- results(dds)
write.csv(as.data.frame(res), file="PN140dnFGFR_results.csv")

#plot fold changes with 2 fold blue line added, alpha is FDR cutoff, change to whatever cutoff you want
pdf("PN140dnFGFR_MAPlot.pdf")
DESeq2::plotMA(res, alpha = 0.1, ylim=c(-3,3), cex=.4)
abline(h=c(-1,1), col="dodgerblue", lwd=2)
dev.off()

```

# PN140 Control effect of housing
```{r}
setwd("/Users/gallifrey/Desktop/dnFGFR/PN140 control/")
#Designate Files
counts1 = read.table("PN140control_counts.txt", header=TRUE, row.names=1, sep="\t")
coldata = read.table("PN140control_coldata.txt", header=TRUE, row.names=1, sep="\t")

#remove length column
counts1['Length'] = NULL

#set dds object, define design
dds = DESeqDataSetFromMatrix(countData = counts1, colData = coldata, design = ~housing)

#remove low count genes (anything less than 20 reads)
keep <- rowSums(counts(dds)) >= 20
dds <- dds[keep,]

#set comparison level, 
dds$housing <- relevel(dds$housing, ref = "same")

#perform normalization, dispersion estimate, and statistical tests
dds = DESeq(dds)

#plot dispersions
pdf("PN140control_dispersionPlot.pdf")
plotDispEsts(dds )
dev.off()

#write out normalized counts
normed = counts(dds, normalized=TRUE)
write.csv(normed, file="PN140controlNormCounts.csv")

#write out statistical results
res <- results(dds)
write.csv(as.data.frame(res), file="PN140control_results.csv")

#plot fold changes with 2 fold blue line added, alpha is FDR cutoff, change to whatever cutoff you want
pdf("PN140control_MAPlot.pdf")
DESeq2::plotMA(res, alpha = 0.1, ylim=c(-3,3), cex=.4)
abline(h=c(-1,1), col="dodgerblue", lwd=2)
dev.off()

```

#Comparing effect of housing in 40D animals controlling for genotype
```{r}
#Designate Files
counts1 = read.table("40D_counts.txt", header=TRUE, row.names=1, sep="\t")
coldata = read.table("40D_coldata.txt", header=TRUE, sep="\t")

#remove length column
counts1['Length'] = NULL

#remove low count genes (anything less than 20 reads)
keep <- rowSums(counts(dds)) >= 20
dds <- dds[keep,]

#set dds object, define design
dds = DESeqDataSetFromMatrix(countData = counts1, colData = coldata, design = ~genotype + housing)

#set comparison level, 
dds$housing <- relevel(dds$housing, ref = "same")

#perform normalization, dispersion estimate, and statistical tests
dds = DESeq(dds)

#plot dispersions
pdf("40D_dispersionPlot.pdf")
plotDispEsts(dds )
dev.off()

#write out normalized counts
normed = counts(dds, normalized=TRUE)
write.csv(normed, file="40D_normCounts.csv")

#write out statistical results
res <- results(dds)
write.csv(as.data.frame(res), file="40D_results.csv")

#plot fold changes with 2 fold blue line added, alpha is FDR cutoff, change to whatever cutoff you want
pdf("40D_MAPlot.pdf")
DESeq2::plotMA(res, alpha = 0.1, ylim=c(-3,3), cex=.4)
abline(h=c(-1,1), col="dodgerblue", lwd=2)
dev.off()

```

#Comparing effect of housing in 120D animals controlling for genotype
```{r}
#Designate Files
counts2 = read.table("120D_counts.txt", header=TRUE, row.names=1, sep="\t")
coldata = read.table("120D_coldata.txt", header=TRUE, sep="\t")

#remove length column
counts2['Length'] = NULL

#remove low count genes (anything less than 20 reads)
keep <- rowSums(counts(dds)) >= 20
dds <- dds[keep,]

#set dds object, define design
dds = DESeqDataSetFromMatrix(countData = counts2, colData = coldata, design = ~genotype + housing)

#set comparison level, 
dds$housing <- relevel(dds$housing, ref = "same")

#perform normalization, dispersion estimate, and statistical tests
dds = DESeq(dds)

#plot dispersions
pdf("120D_dispersionPlot.pdf")
plotDispEsts(dds )
dev.off()

#write out normalized counts
normed = counts(dds, normalized=TRUE)
write.csv(normed, file="120D_normCounts.csv")

#write out statistical results
res <- results(dds)
write.csv(as.data.frame(res), file="120D_results.csv")

#plot fold changes with 2 fold blue line added, alpha is FDR cutoff, change to whatever cutoff you want
pdf("120D_MAPlot.pdf")
DESeq2::plotMA(res, alpha = 0.1, ylim=c(-3,3), cex=.4)
abline(h=c(-1,1), col="dodgerblue", lwd=2)
dev.off()

```


#Comparing effect of genotype on 40D animals while controlling for housing
```{r}
#Designate Files
counts1 = read.table("40D_counts.txt", header=TRUE, row.names=1, sep="\t")
coldata = read.table("40D_coldata.txt", header=TRUE, sep="\t")

#remove length column
counts1['Length'] = NULL

#set dds object, define design
dds = DESeqDataSetFromMatrix(countData = counts1, colData = coldata, design = ~housing + genotype)

#set comparison level, 
#dds$housing <- relevel(dds$housing, ref = "same")

dds$genotype <- relevel(dds$genotype, ref = "ctrl")

#perform normalization, dispersion estimate, and statistical tests
dds = DESeq(dds)

#plot dispersions
pdf("40D_dispersionPlot.pdf")
plotDispEsts(dds )
dev.off()

#write out normalized counts
normed = counts(dds, normalized=TRUE)
write.csv(normed, file="40D_normCounts.csv")

#write out statistical results
res <- results(dds)
write.csv(as.data.frame(res), file="40D_results.csv")

#plot fold changes with 2 fold blue line added, alpha is FDR cutoff, change to whatever cutoff you want
pdf("40D_MAPlot.pdf")
DESeq2::plotMA(res, alpha = 0.1, ylim=c(-3,3), cex=.4)
abline(h=c(-1,1), col="dodgerblue", lwd=2)
dev.off()

```


#Comparing effect of genotype on 120D animals while controlling for housing
```{r}
#Designate Files
counts2 = read.table("120D_counts.txt", header=TRUE, row.names=1, sep="\t")
coldata = read.table("120D_coldata.txt", header=TRUE, sep="\t")

#remove length column
counts2['Length'] = NULL

#remove low count genes (anything less than 20 reads)

#set dds object, define design
dds = DESeqDataSetFromMatrix(countData = counts2, colData = coldata, design = ~ housing + genotype)

#set comparison level, 
dds$genotype <- relevel(dds$genotype, ref = "ctrl")

#perform normalization, dispersion estimate, and statistical tests
dds = DESeq(dds)

#plot dispersions
pdf("120D_dispersionPlot.pdf")
plotDispEsts(dds )
dev.off()

#write out normalized counts
normed = counts(dds, normalized=TRUE)
write.csv(normed, file="120D_normCounts.csv")

#write out statistical results
res <- results(dds)
write.csv(as.data.frame(res), file="120D_results.csv")

#plot fold changes with 2 fold blue line added, alpha is FDR cutoff, change to whatever cutoff you want
pdf("120D_MAPlot.pdf")
DESeq2::plotMA(res, alpha = 0.1, ylim=c(-3,3), cex=.4)
abline(h=c(-1,1), col="dodgerblue", lwd=2)
dev.off()

```


# All samples, effect of genotype controlling for housing and time
```{r}
#Designate Files
setwd("/Users/gallifrey/Desktop/dnFGFR/allsamples/deseq2/genotype/")

counts1 = read.table("all_counts.txt", header=TRUE, row.names=1, sep="\t")
coldata = read.table("all_coldata.txt", header=TRUE, sep="\t")

#remove length column
counts1['Length'] = NULL

#set dds object, define design
dds = DESeqDataSetFromMatrix(countData = counts1, colData = coldata, design = ~ housing + time + genotype)

#remove low count genes (anything less than 20 reads)
keep <- rowSums(counts(dds)) >= 20
dds <- dds[keep,]

#set comparison level, 
dds$housing <- relevel(dds$housing, ref = "same")
dds$genotype <- relevel(dds$genotype, ref = "ctrl")

#perform normalization, dispersion estimate, and statistical tests
dds = DESeq(dds)

#plot dispersions
pdf("all_dispersionPlot.pdf")
plotDispEsts(dds )
dev.off()

#write out normalized counts
normed = counts(dds, normalized=TRUE)
write.csv(normed, file="all_genotype_normCounts.csv")

#write out statistical results
res <- results(dds)
write.csv(as.data.frame(res), file="all_genotype_results.csv")

#plot fold changes with 2 fold blue line added, alpha is FDR cutoff, change to whatever cutoff you want
pdf("all_genotype_MAPlot.pdf")
DESeq2::plotMA(res, alpha = 0.1, ylim=c(-3,3), cex=.4)
abline(h=c(-1,1), col="dodgerblue", lwd=2)
dev.off()

```


# All samples, effect of housing controlling for genotype and time
```{r}
#Designate Files
setwd("/Users/gallifrey/Desktop/dnFGFR/allsamples/deseq2/housing/")

counts1 = read.table("all_counts.txt", header=TRUE, row.names=1, sep="\t")
coldata = read.table("all_coldata.txt", header=TRUE, sep="\t")

#remove length column
counts1['Length'] = NULL

#set dds object, define design
dds = DESeqDataSetFromMatrix(countData = counts1, colData = coldata, design = ~ genotype + time + housing)

#remove low count genes (anything less than 20 reads)
keep <- rowSums(counts(dds)) >= 20
dds <- dds[keep,]

#set comparison level, 
dds$housing <- relevel(dds$housing, ref = "same")
dds$genotype <- relevel(dds$genotype, ref = "ctrl")

#perform normalization, dispersion estimate, and statistical tests
dds = DESeq(dds)

#plot dispersions
pdf("all_housing_dispersionPlot.pdf")
plotDispEsts(dds )
dev.off()

#write out normalized counts
normed = counts(dds, normalized=TRUE)
write.csv(normed, file="all_housing_normCounts.csv")

#write out statistical results
res <- results(dds)
write.csv(as.data.frame(res), file="all_housing_results.csv")

#plot fold changes with 2 fold blue line added, alpha is FDR cutoff, change to whatever cutoff you want
pdf("all_housing_MAPlot.pdf")
DESeq2::plotMA(res, alpha = 0.1, ylim=c(-3,3), cex=.4)
abline(h=c(-1,1), col="dodgerblue", lwd=2)
dev.off()

```


# ----------------------------------------------------------------------

#Example of basic comparison of OppSex vs SameSex in 40D control mice. - From Patrick
```{r}
#read in files
counts1 = read.table("control_40D_counts.txt", header=TRUE, row.names=1, sep="\t")
coldata = read.table("control_40D_coldata.txt", header=TRUE, sep="\t")

#remove length column
counts1['Length'] = NULL

#remove low count genes
keep <- rowSums(counts(dds)) >= 20
dds <- dds[keep,]


#set dds object, define design
dds = DESeqDataSetFromMatrix(countData = counts1, colData = coldata, design = ~housing)


#set comparison level, 
dds$housing <- relevel(dds$housing, ref = "sameSex")

#perform normalization, dispersion estimate, and statistical tests
dds = DESeq(dds)

#plot dispersions
pdf("control_40D_dispersionPlot.pdf")
plotDispEsts(dds )
dev.off()

#write out normalized counts
normed = counts(dds, normalized=TRUE)
write.csv(normed, file="control_40D_normCounts.csv")

#write out statistical results
res <- results(dds)
write.csv(as.data.frame(res), file="control_40D_results.csv")

#plot fold changes with 2 fold blue line added, alpha is FDR cutoff, change to whatever cutoff you want
pdf("control_40D_MAPlot.pdf")
DESeq2::plotMA(res, alpha = 0.1, ylim=c(-3,3), cex=.4)
abline(h=c(-1,1), col="dodgerblue", lwd=2)
dev.off()


```



