---
title: "Heatmaps"
author: "Tyler Akonom"
date: "4/22/2024"
output: html_document
---

```{r setup, include=FALSE}
# Second round of dnFGFR OS sequencing. Animals were either OS or SS housed for 140 or 320 days.
require("knitr")
knitr::opts_chunk$set(echo = TRUE)
opts_knit$set(root.dir = "C:\\GitHub\\dnFGFR\\deseq_outputs\\")
outdir <- "C:\\GitHub\\dnFGFR\\deseq_outputs\\"
library(dplyr)
library(DESeq2)
library(knitr)
library(ggplot2)
library(magrittr)
library(tidyr)
library(tibble)
library(ComplexHeatmap)
#library(DEGreport)
library(vsn)
library(rrcov)
library(sva)
library(circlize)
library(enrichplot)

```


# Load counts and metadata
```{r}
counts = read.csv("Rsub_less_id.coverage.csv", header=TRUE, row.names = 1)
colnames(counts) <- sub(".bam", "", colnames(counts))
coldata = read.table("coldata.txt", header=TRUE, sep="\t")
coldata$group<-paste(coldata$genotype, coldata$time, coldata$housing, sep="_")
coldata$timehousing <- paste(coldata$time, coldata$housing, sep="_")
coldata$housing <- factor(coldata$housing, levels = c("same","opposite"))
coldata$genotype <- factor(coldata$genotype, levels = c("ctrl", "dnFGFR"))
coldata$time <- factor(coldata$time, levels = c("PN140",  "PN320"))
coldata$group <- factor(coldata$group)
coldata$timehousing <- factor(coldata$timehousing, levels = c("PN140_same", "PN140_opposite","PN320_same", "PN320_opposite"))
#coldata$parent <- factor(coldata$parent)
coldata$dod <- factor(coldata$dod)

#remove length from counts table and sort properly
counts['Length'] = NULL
counts <- counts[,coldata$sampleName]

threshold <- 0.05
```


# Investigating whether or not genotypes show a different response to OS housing
```{r}
coldataedit <- coldata %>% filter(time == "PN320")
removesamples <- c("PN140CS2", "PN140CS4", "PN140DS1", "PN320DO5")
coldataedit <- coldataedit %>% filter(!sampleName %in% removesamples)
countedit <- counts %>% select(coldataedit$sampleName)
dds = DESeqDataSetFromMatrix(countData = countedit, colData = coldataedit, design = ~genotype + housing + genotype:housing)

dds = DESeq(dds)
plotDispEsts(dds)
effect_housing_control_res <- results(dds, contrast = c("housing", "opposite", "same"))
effect_housing_dnFGFR_res <- results(dds, list(c("housing_opposite_vs_same", "genotypednFGFR.housingopposite")))

threshold = 0.1
summary(effect_housing_control_res, alpha = threshold)
plotMA(effect_housing_control_res, alpha = threshold)

summary(effect_housing_dnFGFR_res, alpha = threshold)
plotMA(effect_housing_dnFGFR_res, alpha = threshold)

effect_housing_dnFGFR_res[order(effect_housing_dnFGFR_res$padj), ]
effect_housing_control_res[order(effect_housing_control_res$padj), ]
plotCounts(dds, gene = "Drd3", intgroup = c("genotype", "housing"))

hist(effect_housing_control_res$pvalue, breaks = 200)
hist(effect_housing_control_res$padj, breaks = 100)

hist(effect_housing_dnFGFR_res$pvalue, breaks = 200)
hist(effect_housing_dnFGFR_res$padj, breaks = 100)

plotCounts(dds, gene = "Acaa1b", intgroup = c("genotype", "housing"))
```

# Outlier detection using robust PCA

```{r}
countsT <- t(countedit)
pcah <- PcaHubert(countsT)
pcag <- PcaGrid(countsT)
plot(pcag)

#Sample PN320DO5 was eliminated during this step and removed from further analysis.

```


```{r}
# Write out results
setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\effect_housing_dnFGFR_res\\")
write.csv(as.data.frame(effect_housing_dnFGFR_res), file="effect_housing_dnFGFR_res.csv")
 
# Create MA Plot, alpha is FDR cutoff, for this experiment it is "0.05"
pdf("effect_housing_dnFGFR_res_MAPlot.pdf")
DESeq2::plotMA(effect_housing_dnFGFR_res, alpha = 0.05, ylim=c(-3,3), cex=.4)
abline(h=c(-1,1), col="dodgerblue", lwd=2)
dev.off()

# Plot dispersions
pdf("effect_housing_dnFGFR_res_dispersionPlot.pdf")
plotDispEsts(dds)
dev.off()
```

# Generating plotcount graphs for each gene of interest
```{r}
# Creating a PDF of each gene
setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\effect_housing_dnFGFR_res\\plotcounts\\")

genesofinterest <- c("Kiss1", "Kiss1r", "Th", "Gnrhr", "Gnrh1")

goi <- as.data.frame(effect_housing_dnFGFR_res)
goi$genename <- rownames(goi)
goi <- goi %>% filter(genename %in% genesofinterest)

normed = as.data.frame(counts(dds, normalized=TRUE))
normed$genename <- rownames(normed)
normededit <- normed %>% filter(genename %in% genesofinterest)

for (genename in genesofinterest){
  n <- paste(genename,"png", sep = ".")
  png(filename = n, width = 12, height = 12, units = "in", res = 800, bg= "white")
  plotCounts(dds,gene=genename,intgroup= c("genotype", "housing"), normalized = TRUE)
  dev.off()}

data <- plotCounts(dds, 
                   gene = "Kiss1r", 
                   intgroup = c("genotype","housing"), 
                   returnData = TRUE)

ggplot(data, aes(x=interaction(genotype, housing), y=count, color=genotype, shape=housing)) + 
  geom_jitter(size=3) + 
  scale_y_log10()

goi$genename <- NULL
normededit$genename <- NULL
write.csv(as.data.frame(goi), file="genesofinterest_stat.csv")
write.csv(as.data.frame(normededit), file="genesofinterest_normalized_counts.csv")
```


```{r}
outdir <- "C:\\GitHub\\dnFGFR\\deseq_outputs\\effect_housing_dnFGFR_res\\"

effect_housing_dnFGFR_res[order(effect_housing_dnFGFR_res$padj), ]
plotCounts(dds, gene = "Ttc34", intgroup = c("genotype", "housing"))
resSig <- as.data.frame(subset(effect_housing_dnFGFR_res, padj < threshold))
write.table(resSig,paste0(outdir,"/",'effect_housing_dnFGFR_ressig.txt', sep=""), sep="\t",append = FALSE, quote = FALSE)
allresgenes <- na.omit(as.data.frame(effect_housing_dnFGFR_res))
write.table(allresgenes,paste0(outdir,"/",'effect_housing_dnFGFR_res.txt', sep=""), sep="\t",append = FALSE, quote = FALSE)

```

# Generate ranked file for GSEA
```{r}

rnkdf <- tibble(gene = rownames(effect_housing_dnFGFR_res),
				rnk = -log(effect_housing_dnFGFR_res$pvalue) * sign(effect_housing_dnFGFR_res$log2FoldChange)) %>%
	arrange(desc(rnk)) %>% drop_na()

## Write out the table without any additional information
write.table(rnkdf, file = paste0(outdir,"deseq_dnfgfr_res_for_gsea.rnk"),
			append = FALSE, col.names = FALSE, row.names = FALSE,
			quote = FALSE, sep = "\t")

```


# Raw results for heatmaps
```{r}
normed = counts(dds, normalized=TRUE)
# write.csv(normed, file="normCounts.csv")
siggenes <- rownames(effect_housing_dnFGFR_res)
normedsig <- as.data.frame(normed) %>% filter(rownames(normed) %in% siggenes)

rowSigma <- apply(normedsig, 1, sd, na.rm = TRUE)
rowMu <- rowMeans(normedsig, na.rm = TRUE)
zscore_norm_expression <- (normedsig - rowMu) / rowSigma

```






