---
title: "Rsubread"
output: html_notebook
---

##Running Rsubread for counts:

# source /curc/sw/anaconda/default
# conda activate r361
# R

library(Rsubread)
#CD to TMP
TMPDIR = "/gpfs/summit/scratch/tyak9569/"
TMP = "/gpfs/summit/scratch/tyak9569/"
TEMP = "/gpfs/summit/scratch/tyak9569/"

files=c(
'/gpfs/summit/scratch/tyak9569/dnFGFR/aligned/PN140DO1.bam',
'/gpfs/summit/scratch/tyak9569/dnFGFR/aligned/PN140DO2.bam',
'/gpfs/summit/scratch/tyak9569/dnFGFR/aligned/PN140DO3.bam',
'/gpfs/summit/scratch/tyak9569/dnFGFR/aligned/PN140DO4.bam',
'/gpfs/summit/scratch/tyak9569/dnFGFR/aligned/PN140DO5.bam',
'/gpfs/summit/scratch/tyak9569/dnFGFR/aligned/PN140DS1.bam',
'/gpfs/summit/scratch/tyak9569/dnFGFR/aligned/PN140DS2.bam',
'/gpfs/summit/scratch/tyak9569/dnFGFR/aligned/PN140DS3.bam',
'/gpfs/summit/scratch/tyak9569/dnFGFR/aligned/PN140DS4.bam',
'/gpfs/summit/scratch/tyak9569/dnFGFR/aligned/PN140DS5.bam',
'/gpfs/summit/scratch/tyak9569/dnFGFR/aligned/PN140CO1.bam',
'/gpfs/summit/scratch/tyak9569/dnFGFR/aligned/PN140CO2.bam',
'/gpfs/summit/scratch/tyak9569/dnFGFR/aligned/PN140CO3.bam',
'/gpfs/summit/scratch/tyak9569/dnFGFR/aligned/PN140CO4.bam',
'/gpfs/summit/scratch/tyak9569/dnFGFR/aligned/PN140CO5.bam',
'/gpfs/summit/scratch/tyak9569/dnFGFR/aligned/PN140CS1.bam',
'/gpfs/summit/scratch/tyak9569/dnFGFR/aligned/PN140CS2.bam',
'/gpfs/summit/scratch/tyak9569/dnFGFR/aligned/PN140CS3.bam',
'/gpfs/summit/scratch/tyak9569/dnFGFR/aligned/PN140CS4.bam',
'/gpfs/summit/scratch/tyak9569/dnFGFR/aligned/PN140CS5.bam',
'/gpfs/summit/scratch/tyak9569/dnFGFR/aligned/PN320DO1.bam',
'/gpfs/summit/scratch/tyak9569/dnFGFR/aligned/PN320DO2.bam',
'/gpfs/summit/scratch/tyak9569/dnFGFR/aligned/PN320DO3.bam',
'/gpfs/summit/scratch/tyak9569/dnFGFR/aligned/PN320DO4.bam',
'/gpfs/summit/scratch/tyak9569/dnFGFR/aligned/PN320DO5.bam',
'/gpfs/summit/scratch/tyak9569/dnFGFR/aligned/PN320DS1.bam',
'/gpfs/summit/scratch/tyak9569/dnFGFR/aligned/PN320DS2.bam',
'/gpfs/summit/scratch/tyak9569/dnFGFR/aligned/PN320DS3.bam',
'/gpfs/summit/scratch/tyak9569/dnFGFR/aligned/PN320DS4.bam',
'/gpfs/summit/scratch/tyak9569/dnFGFR/aligned/PN320DS5.bam',
'/gpfs/summit/scratch/tyak9569/dnFGFR/aligned/PN320CO1.bam',
'/gpfs/summit/scratch/tyak9569/dnFGFR/aligned/PN320CO2.bam',
'/gpfs/summit/scratch/tyak9569/dnFGFR/aligned/PN320CO3.bam',
'/gpfs/summit/scratch/tyak9569/dnFGFR/aligned/PN320CO4.bam',
'/gpfs/summit/scratch/tyak9569/dnFGFR/aligned/PN320CO5.bam',
'/gpfs/summit/scratch/tyak9569/dnFGFR/aligned/PN320CS1.bam',
'/gpfs/summit/scratch/tyak9569/dnFGFR/aligned/PN320CS2.bam',
'/gpfs/summit/scratch/tyak9569/dnFGFR/aligned/PN320CS3.bam',
'/gpfs/summit/scratch/tyak9569/dnFGFR/aligned/PN320CS4.bam',
'/gpfs/summit/scratch/tyak9569/dnFGFR/aligned/PN320CS5.bam')

## Point to GTF file:

gtf=("/projects/tyak9569/dnFGFR/genome/mm10.refGene.gtf")

## Output the file:

OUT="/gpfs/summit/scratch/tyak9569/dnFGFR/rsubless/"

# Unstranded

coverage=featureCounts(files,
isGTFAnnotationFile = TRUE,
annot.ext = gtf,
GTF.attrType = "gene_id",
allowMultiOverlap = TRUE,
nthreads = 36,
isPairedEnd=TRUE,
strandSpecific = 0)

# write.table(x=data.frame(
# senseunique$annotation[,c("GeneID","Length")],
# senseunique$counts,stringsAsFactors=FALSE),
# file=paste(OUT,"rsubread_senseunique_0.txt"),
# quote=FALSE,sep="\t",row.names=FALSE) 

# Forward Stranded

# senseunique=featureCounts(files,
# isGTFAnnotationFile = TRUE,
# annot.ext = gtf,
# GTF.attrType = "gene_id",
# allowMultiOverlap = FALSE,
# nthreads = 18,
# isPairedEnd=TRUE,
# strandSpecific = 1)

# write.table(x=data.frame(
# senseunique$annotation[,c("GeneID","Length")],
# senseunique$counts,stringsAsFactors=FALSE),
# file=paste(OUT,"rsubread_senseunique_1.txt"),
# quote=FALSE,sep="\t",row.names=FALSE) 

# Reverse Stranded

# senseunique=featureCounts(files,
# isGTFAnnotationFile = TRUE,
# annot.ext = gtf,
# GTF.attrType = "gene_id",
# allowMultiOverlap = FALSE,
# nthreads = 18,
# isPairedEnd=TRUE,
# strandSpecific = 2)

# write.table(x=data.frame(
# senseunique$annotation[,c("GeneID","Length")],
# senseunique$counts,stringsAsFactors=FALSE),
# file=paste(OUT,"rsubread_senseunique_2.txt"),
# quote=FALSE,sep="\t",row.names=FALSE) 

# Antisense

# coverage=featureCounts(files,
# isGTFAnnotationFile = TRUE,
# annot.ext = gtf,
# GTF.attrType = "gene_id",
# allowMultiOverlap = TRUE,
# isPairedEnd=FALSE,
# nthreads = 18,
# strandSpecific = 2)
 
# write.table(x=data.frame(
# antisenseunique$annotation[,c("GeneID","Length")],
# antisenseunique$counts,stringsAsFactors=FALSE),
# file=paste(OUT,"Rsubread_antisenseunique"),
# quote=FALSE,sep="\t",row.names=FALSE)


fileroot=paste(OUT,"Rsub_less_id", sep="")
write.csv(coverage$counts, paste(fileroot,".coverage.csv", sep=""))
write.csv(coverage$stat, paste(fileroot,".stat.csv", sep=""))
write.csv(coverage$annotation, paste(fileroot,".annotation.csv", sep=""))
write.csv(coverage$targets, paste(fileroot,".targets.csv", sep=""))

