---
title: "R Notebook"
output: html_notebook
---

```{r setup, include=FALSE}
# Second round of dnFGFR OS sequencing. Animals were either OS or SS housed for 140 or 320 days.
require("knitr")
knitr::opts_chunk$set(echo = TRUE)
opts_knit$set(root.dir = "C:\\GitHub\\dnFGFR\\deseq_outputs\\")
library(dplyr)
library(DESeq2)
library(knitr)
library(ggplot2)
library(magrittr)
library(tidyr)
library(tibble)
library(DEGreport)
library(vsn)
library(rrcov)
library(sva)
```


# Load counts and metadata
```{r}
counts = read.csv("Rsubread.coverage.csv", header=TRUE, row.names = 1)
colnames(counts) <- sub(".bam", "", colnames(counts))
coldata = read.table("coldata.txt", header=TRUE, sep="\t")
coldata$group<-paste(coldata$genotype, coldata$time, coldata$housing, sep="_")
coldata$timehousing <- paste(coldata$time, coldata$housing, sep="_")
coldata$housing <- factor(coldata$housing, levels = c("same","opposite"))
coldata$genotype <- factor(coldata$genotype, levels = c("ctrl", "dnFGFR"))
coldata$time <- factor(coldata$time, levels = c("PN140",  "PN320"))
coldata$group <- factor(coldata$group)
coldata$timehousing <- factor(coldata$timehousing, levels = c("PN140_same", "PN140_opposite","PN320_same", "PN320_opposite"))
#coldata$parent <- factor(coldata$parent)
coldata$dod <- factor(coldata$dod)

#remove length from counts table and sort properly
counts['Length'] = NULL
counts <- counts[,coldata$sampleName]

threshold <- 0.05
```


```{r}
coldataedit <- coldata %>% filter(time == "PN320")
removesamples <- c("PN140CS2", "PN320DO5")
coldataedit <- coldataedit %>% filter(!sampleName %in% removesamples)
countedit <- counts %>% select(coldataedit$sampleName)
dds = DESeqDataSetFromMatrix(countData = countedit, colData = coldataedit, design = ~genotype + housing + genotype:housing)

dds = DESeq(dds)
plotDispEsts(dds)
effect_housing_control_res <- results(dds, contrast = c("housing", "opposite", "same"))
effect_housing_dnFGFR_res <- results(dds, list(c("housing_opposite_vs_same", "genotypednFGFR.housingopposite")))

threshold = 0.01
summary(effect_housing_control_res, alpha = threshold)
plotMA(effect_housing_control_res, alpha = threshold)

summary(effect_housing_dnFGFR_res, alpha = threshold)
plotMA(effect_housing_dnFGFR_res, alpha = threshold)

effect_housing_dnFGFR_res[order(effect_housing_dnFGFR_res$padj), ]
plotCounts(dds, gene = "Otof", intgroup = c("genotype", "housing"))

hist(effect_housing_dnFGFR_res$pvalue, breaks = 200)
hist(effect_housing_dnFGFR_res$padj, breaks = 100)
```