---
title: "dnFGFR"
author: "Tyler Akonom"
date: "6/20/22"
output: html_document
---

```{r setup, include=FALSE}
# Second round of dnFGFR OS sequencing. Animals were either OS or SS housed for 140 or 320 days.
require("knitr")
knitr::opts_chunk$set(echo = TRUE)
opts_knit$set(root.dir = "C:\\GitHub\\dnFGFR\\deseq_outputs\\")
library(dplyr)
library(DESeq2)
library(knitr)
library(ggplot2)
library(magrittr)
library(tidyr)
library(tibble)
library(DEGreport)
library(vsn)
library(rrcov)
library(sva)
```


# Load counts and metadata
```{r}
counts = read.csv("Rsub_less_id.coverage.csv", header=TRUE, row.names = 1)
colnames(counts) <- sub(".bam", "", colnames(counts))
coldata = read.table("coldata.txt", header=TRUE, sep="\t")
coldata$group<-paste(coldata$genotype, coldata$time, coldata$housing, sep="_")
coldata$timehousing <- paste(coldata$time, coldata$housing, sep="_")
coldata$housing <- factor(coldata$housing, levels = c("same","opposite"))
coldata$genotype <- factor(coldata$genotype, levels = c("ctrl", "dnFGFR"))
coldata$time <- factor(coldata$time, levels = c("PN140",  "PN320"))
coldata$group <- factor(coldata$group)
coldata$timehousing <- factor(coldata$timehousing, levels = c("PN140_same", "PN140_opposite","PN320_same", "PN320_opposite"))
#coldata$parent <- factor(coldata$parent)
coldata$dod <- factor(coldata$dod)

#remove length from counts table and sort properly
counts['Length'] = NULL
counts <- counts[,coldata$sampleName]

threshold <- 0.1
```

# Outlier detection using robust PCA

```{r}
countsT <- t(countedit)
pcah <- PcaHubert(countsT)
pcag <- PcaGrid(countsT)
plot(pcag)
```


# Checking to see if parent cage or date of collection drives expression

```{r}
dds = DESeqDataSetFromMatrix(countData = counts, colData = coldata, design = ~time)
ntd <- normTransform(dds)
meanSdPlot(assay(ntd))
sampleDists <- dist(t(assay(ntd)))
# pdf("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\PCA\\PCA_DOD.pdf")
plotPCA(ntd, intgroup=c("genotype"))
# dev.off()


coldataedit <- coldata %>% filter(time == "PN320")
countedit <- counts %>% select(coldataedit$sampleName)
dds = DESeqDataSetFromMatrix(countData = countedit, colData = coldataedit, design = ~genotype + housing)
ntd <- normTransform(dds)
meanSdPlot(assay(ntd))
sampleDists <- dist(t(assay(ntd)))
# pdf("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\PCA\\PCA_DOD.pdf")
plotPCA(ntd, intgroup=c("dod"))
# dev.off()

svacounts <- ComBat_seq(counts(dds), batch = coldata$time)

svadds = DESeqDataSetFromMatrix(countData = svacounts, colData = coldata, design = ~housing)
svantd <- normTransform(svadds)
meanSdPlot(assay(svantd))
sampleDists <- dist(t(assay(svantd)))
# pdf("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\PCA\\PCA_DOD.pdf")
plotPCA(svantd, intgroup=c("housing"))
# dev.off()

svacountsT <- t(svacounts)
svapcag <- PcaGrid(svacountsT)
plot(svapcag)

# Create DESeq object
removesamples <- c("PN140CS2", "PN320DO5")
coldataedit <- coldata %>% filter(!sampleName %in% removesamples)
svacountedit <- as.data.frame(svacounts) %>% select(coldataedit$sampleName)

svadds = DESeqDataSetFromMatrix(countData = svacountedit, colData = coldataedit, design = ~ time + genotype + housing)

# Run DESeq on the DESeq object
svadds = DESeq(svadds)
plotDispEsts(svadds)
svares <- results(svadds, contrast = c("housing", "opposite", "same"))
summary(svares, alpha = threshold)
plotMA(svares, alpha = threshold)

svadds_lrt <- DESeq(svadds, test = "LRT", reduced = ~1)
svalrtres <- results(svadds_lrt)
summary(svalrtres, alpha = threshold)
plotMA(svalrtres, alpha = threshold)

esvadds_lrt <- DESeq(svadds, test = "LRT", reduced = ~time + genotype)
esvalrtres <- results(esvadds_lrt)
summary(esvalrtres, alpha = threshold)
plotMA(esvalrtres, alpha = threshold)

esvalrtres[order(esvalrtres$padj), ]
plotCounts(esvadds_lrt, gene = "ENSMUSG00000048978", intgroup = c("time", "genotype", "housing"))

```



```{r}
coldataedit <- coldata %>% filter(time == "PN320")
removesamples <- c("PN140CS2", "PN140CS4", "PN140DS1", "PN320DO5")
coldataedit <- coldata %>% filter(!sampleName %in% removesamples)
countedit <- counts %>% select(coldataedit$sampleName)
dds = DESeqDataSetFromMatrix(countData = countedit, colData = coldataedit, design = ~genotype + housing + genotype:housing)

dds = DESeq(dds)
plotDispEsts(dds)
effect_housing_control_res <- results(dds, contrast = c("housing", "opposite", "same"))
effect_housing_dnFGFR_res <- results(dds, list(c("housing_opposite_vs_same", "genotypednFGFR.housingopposite")))

threshold = 0.01
summary(effect_housing_control_res, alpha = threshold)
plotMA(effect_housing_control_res, alpha = threshold)

summary(effect_housing_dnFGFR_res, alpha = threshold)
plotMA(effect_housing_dnFGFR_res, alpha = threshold)

effect_housing_dnFGFR_res[order(effect_housing_dnFGFR_res$padj), ]
plotCounts(dds, gene = "ENSMUSG00000038503", intgroup = c("genotype", "housing"))

hist(effect_housing_dnFGFR_res$pvalue, breaks = 200)
hist(effect_housing_dnFGFR_res$padj, breaks = 100)
```




```{r}
dds = DESeqDataSetFromMatrix(countData = counts, colData = coldata, design = ~ genotype + timehousing)
ntd <- normTransform(dds)
meanSdPlot(assay(ntd))
sampleDists <- dist(t(assay(ntd)))
# pdf("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\PCA\\PCA_DOD.pdf")
plotPCA(ntd, intgroup=c("group"))
dev.off()
```


```{r}
# dds = DESeqDataSetFromMatrix(countData = counts, colData = coldata, design = ~parent)
# ntd <- normTransform(dds)
# meanSdPlot(assay(ntd))
# sampleDists <- dist(t(assay(ntd)))
# pdf("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\PCA\\PCA_Parent.pdf")
# plotPCA(ntd, intgroup=c("parent"))
# dev.off()
```

# Creating a DESeq object and check for differential expression ------------------------------------------------------

```{r}
setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\")

# Create DESeq object
removesamples <- c("PN140CS2", "PN320DO5")
coldataedit <- coldata %>% filter(!sampleName %in% removesamples)
countedit <- counts %>% select(coldataedit$sampleName)

dds = DESeqDataSetFromMatrix(countData = countedit, colData = coldataedit, design = ~ time + genotype + housing)

# Run DESeq on the DESeq object
dds = DESeq(dds)
plotDispEsts(dds)
res <- results(dds, contrast = c("housing", "opposite", "same"))
summary(res, alpha = threshold)
plotMA(res, alpha = threshold)

normed = counts(dds, normalized=TRUE)
resSig <- subset(res, padj < threshold)
rs <- as.data.frame(resSig)
siggenes <- as.vector(rownames(rs))

# # Generating a .rnk file for GSEA from results
# rnkdf <- tibble(gene = rownames(res),
# 				rnk = -log(res$pvalue) * sign(res$log2FoldChange)) %>%
# 	arrange(desc(rnk)) %>% drop_na()
#
## Write out the table without any additional information
# write.table(rnkdf, file = "deseq_res_for_gsea.rnk", append = FALSE, col.names = FALSE, row.names = FALSE, quote = FALSE, sep = "\t")
#
# plotCounts(dds, gene = "Gm11428", intgroup = "timehousing")
#
# # Creating a PDF of each gene identified above
# setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\figures\\")
# for (genename in siggenes){
#   n <- paste(genename,"png", sep = ".")
#   png(filename = n, width = 12, height = 12, units = "in", res = 800, bg= "NA")
#   plotCounts(dds,gene=genename,intgroup= "timehousing")
#   dev.off()}

# Write out results
write.csv(as.data.frame(res), file="timehousing-genotype_results.csv")
 
# Create MA Plot, alpha is FDR cutoff, for this experiment it is "0.05"
pdf("timehousing-genotype_MAPlot.pdf")
DESeq2::plotMA(res, alpha = 0.05, ylim=c(-3,3), cex=.4)
abline(h=c(-1,1), col="dodgerblue", lwd=2)
dev.off()

# Write out normalized counts
write.csv(normed, file="normCounts.csv")

# Plot dispersions
pdf("timehousing-genotype_dispersionPlot.pdf")
plotDispEsts(dds)
dev.off()
```

```{r}
setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\")

# Create DESeq object
removesamples <- c("PN140CS2", "PN320DO5")
coldataedit <- coldata %>% filter(!sampleName %in% removesamples)
countedit <- counts %>% select(coldataedit$sampleName)

dds = DESeqDataSetFromMatrix(countData = countedit, colData = coldataedit, design = ~ time + genotype + housing)

DEdds <- DESeq(dds, test="LRT", reduced = ~ 1)

#plot the dispersion of the data
plotDispEsts(DEdds)

#get the results dataframes (which was not )
res_LRT <- results(DEdds)

#look at and output the results of one comparison
res = res_LRT
summary(res)
plotMA(res, alpha = threshold)
resSig <- subset(res, padj < threshold)
head(res[ order( res$padj ), ])

sig_res_LRT <- res_LRT %>%
  data.frame() %>%
  rownames_to_column(var="gene") %>% 
  as_tibble() %>% 
  filter(padj < threshold)

# Get sig gene lists
sigLRT_genes <- sig_res_LRT %>% 
  pull(gene)

length(sigLRT_genes)

#clustering takes a while so we are only going to cluster on the top 1000 genes in class
#make sure to use all genes when running this for real
clustering_sig_genes <- sig_res_LRT %>%
  arrange(padj) %>%
  head(n=1000)

# Obtain rlog values for those significant genes
rld <- rlog(DEdds, blind=TRUE)
rld_mat <- assay(rld)
cluster_rlog <- rld_mat[clustering_sig_genes$gene, ]

#make a metadata frame for this anaysis... different becuase the index must be the column names for the counts
meta <-as.data.frame(cbind(paste(DEdds$sampleName), paste(DEdds$time), paste(DEdds$genotype),paste(DEdds$housing)))
colnames(meta)<-c("sampleName", "time", "genotype", "housing")
row.names(meta)<-meta$sampleName
meta$genotype <- factor(meta$genotype)
meta$time <- factor(meta$time)
meta$housing <- factor(meta$housing, levels = c("same", "opposite"))

#dev.off()
#jpeg(paste0(outdir,outfilename,"/",'clusters.jpg', sep=""))
clusters <- degPatterns(cluster_rlog, metadata = meta, time = "housing", col= "genotype")
#dev.off()


```



```{r}
#Generate Heatmap of DEGs

mininorm <- normed[noquote(rownames(normed)) %in% noquote(as.vector(siggenes)),]
  mininorm <- as.matrix(mininorm)
  setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\")
  title = "Differentially Expressed Genes"
  png(filename = "deg_heatmap.png", width = 12, height = 12, units = "in", res = 800, bg= "NA")
  heatmap(mininorm, Colv = NA, main=title)
  dev.off()
  
```



```{r}
setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\housing\\")

# Create DESeq object
dds = DESeqDataSetFromMatrix(countData = counts, colData = coldata, design = ~ time + genotype + housing)

# Run DESeq on the DESeq object
dds = DESeq(dds)
res <- results(dds, contrast = c("timehousing", "PN320_opposite", "PN320_same"))
normed = counts(dds, normalized=TRUE)
resSig <- subset(res, padj < 0.05)
rs <- as.data.frame(resSig)
siggenes <- as.vector(rownames(rs))

# Write out results
setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\housing\\")
write.csv(as.data.frame(res), file="dn140_results.csv")

# Create MA Plot, alpha is FDR cutoff, for this experiment it is "0.05"
setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\housing\\")
pdf("housing_MAPlot.pdf")
DESeq2::plotMA(res, alpha = 0.05, ylim=c(-3,3), cex=.4)
abline(h=c(-1,1), col="dodgerblue", lwd=2)
dev.off()

# Write out normalized counts
write.csv(normed4, file="all_normCounts.csv")

# Plot dispersions
pdf("housing_dispersionPlot.pdf")
plotDispEsts(dds4)
dev.off()
```

# GSEA Results -----------------------------------------------------------------------------------------------------------------


```{r}
gsea_inflammatory <- read.table("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\GSEA\\HALLMARK_INFLAMMATORY_RESPONSE.tsv", sep = "\t", row.names = NULL, header = TRUE)

mininormed <- normed[noquote(toupper(as.character(rownames(normed)))) %in% noquote(unique(gsea_inflammatory$SYMBOL)),]
  mininormed <- as.matrix(mininormed)
  setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\GSEA\\")
  title = "Hallmark Inflammatory Response"
  png(filename = "inflammatory_response_w.png", width = 12, height = 12, units = "in", res = 800, bg= "white")
  heatmap(mininormed, Colv = NA, main=title)
  dev.off()
  
minisig140 <- select(as.data.frame(normed), contains("B"))
  minisig140 <- minisig140[noquote(toupper(as.character(rownames(minisig140)))) %in% noquote(unique(gsea_inflammatory$SYMBOL)),]
  minisig140 <- minisig140[noquote(rownames(minisig140)) %in% noquote(unique(siggenes)),]
  minisig140 <- as.matrix(minisig140)
  setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\GSEA\\")
  title = "Hallmark Inflammatory Response"
  png(filename = "inflammatory_response_sig.png", width = 12, height = 12, units = "in", res = 800, bg= "white")
  heatmap(minisig140, Colv = NA, main=title)
  dev.off()
  
gsea_inta <- read.table("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\GSEA\\HALLMARK_INTERFERON_ALPHA_RESPONSE.tsv", sep = "\t", row.names = NULL, header = TRUE)

mininormed <- normed[noquote(toupper(as.character(rownames(normed)))) %in% noquote(unique(gsea_inta$SYMBOL)),]
  mininormed <- as.matrix(mininormed)
  setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\GSEA\\")
  title = "Hallmark Interferon Alpha Response"
  png(filename = "interferon_alpha_response.png", width = 12, height = 12, units = "in", res = 800, bg= "white")
  heatmap(mininormed, Colv = NA, main=title)
  dev.off()
  
minisig140 <- select(as.data.frame(normed), contains("B"))
  minisig140 <- minisig140[noquote(toupper(as.character(rownames(minisig140)))) %in% noquote(unique(gsea_inta$SYMBOL)),]
  minisig140 <- minisig140[noquote(rownames(minisig140)) %in% noquote(unique(siggenes)),]
  minisig140 <- as.matrix(minisig140)
  setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\GSEA\\")
  title = "Hallmark Interferon Alpha Response"
  png(filename = "interferon_alpha_response_sig.png", width = 12, height = 12, units = "in", res = 800, bg= "white")
  heatmap(minisig140, Colv = NA, main=title)
  dev.off()

```


# GO Term Heatmaps ----------------------------------------------------------------------------------------------------------------


```{r}
go_procp <- read.table("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\Go_heatmaps\\positive_regulation_of_cytokine_production.txt", sep = "\t", row.names = NULL, header = TRUE, quote = "")

# normed140 <- as.data.frame(normed)
# normed140 <- select(normed140, contains("B"))
# normed140 <- log(normed140[!(apply(normed140, 1, function(y) any(y == 0))),])

mininormed <- normed[noquote(rownames(normed)) %in% noquote(unique(as.vector(go_procp$MGI.Gene.Marker.ID))),]
  mininormed <- as.matrix(mininormed)
  setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\Go_heatmaps\\")
  title = "Positive Regulation of Cytokine Production"
  png(filename = "positive_regulation_of_cytokine_production.png", width = 12, height = 12, units = "in", res = 800, bg= "NA")
  heatmap(mininormed, Colv = NA, main=title)
  dev.off()
  
minisig140 <- select(as.data.frame(normed), contains("B"))
  minisig140 <- minisig140[noquote(rownames(minisig140)) %in% noquote(unique(as.vector(go_procp$MGI.Gene.Marker.ID))),]
  minisig140 <- minisig140[noquote(rownames(minisig140)) %in% noquote(unique(siggenes)),]
  minisig140 <- as.matrix(minisig140)
  setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\Go_heatmaps\\")
  title = "Positive Regulation of Cytokine Production"
  png(filename = "positive_regulation_of_cytokine_production_sig.png", width = 12, height = 12, units = "in", res = 800, bg= "NA")
  heatmap(minisig140, Colv = NA, main=title)
  dev.off()

minisig60 <- select(as.data.frame(normed), contains("A"))
  minisig60 <- minisig60[noquote(rownames(minisig60)) %in% noquote(unique(as.vector(go_procp$MGI.Gene.Marker.ID))),]
  minisig60 <- minisig60[noquote(rownames(minisig60)) %in% noquote(unique(siggenes)),]
  minisig60 <- as.matrix(minisig60)
  setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\Go_heatmaps\\")
  title = "Positive Regulation of Cytokine Production"
  png(filename = "positive_regulation_of_cytokine_production.png_sig60", width = 12, height = 12, units = "in", res = 800, bg= "NA")
  heatmap(minisig60, Colv = NA, main=title)
  dev.off()

```


```{r}
go_iir <- read.table("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\Go_heatmaps\\innate_immune_response.txt", sep = "\t", row.names = NULL, header = TRUE, quote = "")

mininormed <- normed[noquote(rownames(normed)) %in% noquote(unique(as.vector(go_iir$MGI.Gene.Marker.ID))),]
  mininormed <- as.matrix(mininormed)
  setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\Go_heatmaps\\")
  title = "Innate Immune Response"
  png(filename = "innate_immune_response.png", width = 12, height = 12, units = "in", res = 800, bg= "NA")
  heatmap(mininormed, Colv = NA, main=title)
  dev.off()
  
minisig140 <- select(as.data.frame(normed), contains("B"))
  minisig140 <- minisig140[noquote(rownames(minisig140)) %in% noquote(unique(as.vector(go_iir$MGI.Gene.Marker.ID))),]
  minisig140 <- minisig140[noquote(rownames(minisig140)) %in% noquote(unique(siggenes)),]
  minisig140 <- as.matrix(minisig140)
  setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\Go_heatmaps\\")
  title = "Innate Immune Response"
  png(filename = "innate_immune_response_sig.png", width = 12, height = 12, units = "in", res = 800, bg= "NA")
  heatmap(minisig140, Colv = NA, main=title)
  dev.off()

minisig60 <- select(as.data.frame(normed), contains("A"))
  minisig60 <- minisig60[noquote(rownames(minisig60)) %in% noquote(unique(as.vector(go_iir$MGI.Gene.Marker.ID))),]
  minisig60 <- minisig60[noquote(rownames(minisig60)) %in% noquote(unique(siggenes)),]
  minisig60 <- as.matrix(minisig60)
  setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\Go_heatmaps\\")
  title = "Innate Immune Response"
  png(filename = "innate_immune_response_sig60.png", width = 12, height = 12, units = "in", res = 800, bg= "NA")
  heatmap(minisig60, Colv = NA, main=title)
  dev.off()

```


```{r}
go_ctrbs <- read.table("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\Go_heatmaps\\cellular_response_to_biotic_stimulus.txt", sep = "\t", row.names = NULL, header = TRUE, quote = "")

mininormed <- normed[noquote(rownames(normed)) %in% noquote(unique(as.vector(go_ctrbs$MGI.Gene.Marker.ID))),]
  mininormed <- as.matrix(mininormed)
  setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\Go_heatmaps\\")
  title = "Cellular Response to Biotic Stimulus"
  png(filename = "cellular_response_to_biotic_stimulus.png", width = 12, height = 12, units = "in", res = 800, bg= "NA")
  heatmap(mininormed, Colv = NA, main=title)
  dev.off()
  
minisig140 <- select(as.data.frame(normed), contains("B"))
  minisig140 <- minisig140[noquote(rownames(minisig140)) %in% noquote(unique(as.vector(go_crtbs$MGI.Gene.Marker.ID))),]
  minisig140 <- minisig140[noquote(rownames(minisig140)) %in% noquote(unique(siggenes)),]
  minisig140 <- as.matrix(minisig140)
  setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\Go_heatmaps\\")
  title = "Cellular Response to Biotic Stimulus"
  png(filename = "cellular_response_to_biotic_stimulus_sig.png", width = 12, height = 12, units = "in", res = 800, bg= "NA")
  heatmap(minisig140, Colv = NA, main=title)
  dev.off()

minisig60 <- select(as.data.frame(normed), contains("A"))
  minisig60 <- minisig60[noquote(rownames(minisig60)) %in% noquote(unique(as.vector(go_crtbs$MGI.Gene.Marker.ID))),]
  minisig60 <- minisig60[noquote(rownames(minisig60)) %in% noquote(unique(siggenes)),]
  minisig60 <- as.matrix(minisig60)
  setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\Go_heatmaps\\")
  title = "Cellular Response to Biotic Stimulus"
  png(filename = "cellular_response_to_biotic_stimulus_sig60.png", width = 12, height = 12, units = "in", res = 800, bg= "NA")
  heatmap(minisig60, Colv = NA, main=title)
  dev.off()

  # write.table(normcounts_group,paste0(outdir,outfilename,"/",'cluster',i,'.txt', sep=""), sep="\t",append = FALSE, quote = FALSE)}

```

```{r}
go_air <- read.table("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\Go_heatmaps\\adaptive_immune_response.txt", sep = "\t", row.names = NULL, header = TRUE, quote = "")

mininormed <- normed[noquote(rownames(normed)) %in% noquote(unique(as.vector(go_air$MGI.Gene.Marker.ID))),]
  mininormed <- as.matrix(mininormed)
  setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\Go_heatmaps\\")
  title = "Adaptive Immune Response"
  png(filename = "adaptive_immune_response.png", width = 12, height = 12, units = "in", res = 800, bg= "white")
  heatmap(mininormed, Colv = NA, main=title)
  dev.off()
  
minisig140 <- select(as.data.frame(normed), contains("B"))
  minisig140 <- minisig140[noquote(rownames(minisig140)) %in% noquote(unique(as.vector(go_air$MGI.Gene.Marker.ID))),]
  minisig140 <- minisig140[noquote(rownames(minisig140)) %in% noquote(unique(siggenes)),]
  minisig140 <- as.matrix(minisig140)
  setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\Go_heatmaps\\")
  title = "Adaptive Immune Response"
  png(filename = "adaptive_immune_response_sig.png", width = 12, height = 12, units = "in", res = 800, bg= "white")
  heatmap(minisig140, Colv = NA, main=title)
  dev.off()

minisig60 <- select(as.data.frame(normed), contains("A"))
  minisig60 <- minisig60[noquote(rownames(minisig60)) %in% noquote(unique(as.vector(go_air$MGI.Gene.Marker.ID))),]
  minisig60 <- minisig60[noquote(rownames(minisig60)) %in% noquote(unique(siggenes)),]
  minisig60 <- as.matrix(minisig60)
  setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\Go_heatmaps\\")
  title = "Adaptive Immune Response"
  png(filename = "adaptive_immune_response_sig60.png", width = 12, height = 12, units = "in", res = 800, bg= "NA")
  heatmap(minisig60, Colv = NA, main=title)
  dev.off()

  # write.table(normcounts_group,paste0(outdir,outfilename,"/",'cluster',i,'.txt', sep=""), sep="\t",append = FALSE, quote = FALSE)}

```
---
title: "DESeq2.v2"
author: "Tyler Akonom"
date: "3/9/21"
output: html_document
---

```{r setup, include=FALSE}
require("knitr")
knitr::opts_chunk$set(echo = TRUE)
opts_knit$set(root.dir = "C:\\GitHub\\dnFGFR\\deseq_outputs\\")
library(dplyr)
library(DESeq2)
library(knitr)
library(ggplot2)
library(magrittr)
library(tidyr)
library(tibble)
library(DEGreport)
library(vsn)
```


# Load counts and metadata
```{r}
counts = read.table("refcounts.txt", header=TRUE, row.names=1, sep="\t")
coldata = read.table("coldata.txt", header=TRUE, sep="\t")
coldata$group<-paste(coldata$genotype, coldata$time, coldata$housing, sep="_")
coldata$timehousing <- paste(coldata$time, coldata$housing, sep="_")
coldata$housing <- factor(coldata$housing, levels = c("same","opposite"))
coldata$genotype <- factor(coldata$genotype, levels = c("ctrl", "dnFGFR"))
coldata$time <- factor(coldata$time, levels = c("PN60",  "PN140"))
coldata$group <- factor(coldata$group)
coldata$timehousing <- factor(coldata$timehousing, levels = c("PN60_same", "PN60_opposite","PN140_same", "PN140_opposite"))
coldata$parent <- factor(coldata$parent)
coldata$dod <- factor(coldata$dod)

#remove length from counts table and sort properly
counts['Length'] = NULL
counts <- counts[,coldata$sampleName]
```


# Checking to see if parent cage or date of collection drives expression
```{r}
# dds = DESeqDataSetFromMatrix(countData = counts, colData = coldata, design = ~dod)
# ntd <- normTransform(dds)
# meanSdPlot(assay(ntd))
# sampleDists <- dist(t(assay(ntd)))
# pdf("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\PCA\\PCA_DOD.pdf")
# plotPCA(ntd, intgroup=c("dod"))
# dev.off()
```

```{r}
# dds = DESeqDataSetFromMatrix(countData = counts, colData = coldata, design = ~parent)
# ntd <- normTransform(dds)
# meanSdPlot(assay(ntd))
# sampleDists <- dist(t(assay(ntd)))
# pdf("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\PCA\\PCA_Parent.pdf")
# plotPCA(ntd, intgroup=c("parent"))
# dev.off()
```

# Creating a DESeq object and check for differential expression ------------------------------------------------------

```{r}
setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\")

# Create DESeq object
dds = DESeqDataSetFromMatrix(countData = counts, colData = coldata, design = ~ genotype + timehousing)

# Run DESeq on the DESeq object
dds = DESeq(dds)
# plotDispEsts(dds)
res <- results(dds, contrast = c("timehousing", "PN140_opposite", "PN140_same"))
normed = counts(dds, normalized=TRUE)
resSig <- subset(res, padj < 0.05)
rs <- as.data.frame(resSig)
siggenes <- as.vector(rownames(rs))

# # Generating a .rnk file for GSEA from results
# rnkdf <- tibble(gene = rownames(res),
# 				rnk = -log(res$pvalue) * sign(res$log2FoldChange)) %>%
# 	arrange(desc(rnk)) %>% drop_na()
#
## Write out the table without any additional information
# write.table(rnkdf, file = "deseq_res_for_gsea.rnk", append = FALSE, col.names = FALSE, row.names = FALSE, quote = FALSE, sep = "\t")
#
plotCounts(dds, gene = "Gm11428", intgroup = "timehousing")
#
# # Creating a PDF of each gene identified above
# setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\figures\\")
# for (genename in siggenes){
#   n <- paste(genename,"png", sep = ".")
#   png(filename = n, width = 12, height = 12, units = "in", res = 800, bg= "NA")
#   plotCounts(dds,gene=genename,intgroup= "timehousing")
#   dev.off()}
#
# # Write out results
# write.csv(as.data.frame(res), file="timehousing-genotype_results.csv")
# 
# # Create MA Plot, alpha is FDR cutoff, for this experiment it is "0.05"
# pdf("timehousing-genotype_MAPlot.pdf")
# DESeq2::plotMA(res, alpha = 0.05, ylim=c(-3,3), cex=.4)
# abline(h=c(-1,1), col="dodgerblue", lwd=2)
# dev.off()
# 
# # Write out normalized counts
# write.csv(normed, file="normCounts.csv")
# 
# # Plot dispersions
# pdf("timehousing-genotype_dispersionPlot.pdf")
# plotDispEsts(dds)
# dev.off()
```


```{r}
setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\pn60\\")

# Create DESeq object
dds = DESeqDataSetFromMatrix(countData = counts, colData = coldata, design = ~ genotype + timehousing)

# Run DESeq on the DESeq object
dds = DESeq(dds)
# plotDispEsts(dds)
res <- results(dds, contrast = c("timehousing", "PN60_opposite", "PN60_same"))
normed = counts(dds, normalized=TRUE)
resSig <- subset(res, padj < 0.05)
rs <- as.data.frame(resSig)
siggenes <- as.vector(rownames(rs))

# # Generating a .rnk file for GSEA from results
# rnkdf <- tibble(gene = rownames(res),
# 				rnk = -log(res$pvalue) * sign(res$log2FoldChange)) %>%
# 	arrange(desc(rnk)) %>% drop_na()
#
## Write out the table without any additional information
# write.table(rnkdf, file = "deseq_res_for_gsea.rnk", append = FALSE, col.names = FALSE, row.names = FALSE, quote = FALSE, sep = "\t")
#
plotCounts(dds, gene = "Gpnmb", intgroup = "timehousing")
#
# Creating a PDF of each gene identified above
setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\pn60\\figures\\")
for (genename in siggenes){
  n <- paste(genename,"png", sep = ".")
  png(filename = n, width = 12, height = 12, units = "in", res = 800, bg= "white")
  plotCounts(dds,gene=genename,intgroup= "timehousing")
  dev.off()}

# Write out results
write.csv(as.data.frame(res), file="timehousing-genotype_results.csv")

# Create MA Plot, alpha is FDR cutoff, for this experiment it is "0.05"
pdf("timehousing-genotype_MAPlot.pdf")
DESeq2::plotMA(res, alpha = 0.05, ylim=c(-3,3), cex=.4)
abline(h=c(-1,1), col="dodgerblue", lwd=2)
dev.off()

# Write out normalized counts
write.csv(normed, file="normCounts.csv")

# Plot dispersions
pdf("timehousing-genotype_dispersionPlot.pdf")
plotDispEsts(dds)
dev.off()
```

```{r}
#Generate Heatmap of DEGs

mininorm <- normed[noquote(rownames(normed)) %in% noquote(as.vector(siggenes)),]
  mininorm <- as.matrix(mininorm)
  setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\")
  title = "Differentially Expressed Genes"
  png(filename = "deg_heatmap.png", width = 12, height = 12, units = "in", res = 800, bg= "NA")
  heatmap(mininorm, Colv = NA, main=title)
  dev.off()
  
```



```{r}
setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\dn140\\")

# Create DESeq object
dds = DESeqDataSetFromMatrix(countData = counts, colData = coldata, design = ~group)

# Run DESeq on the DESeq object
dds = DESeq(dds)
res <- results(dds, contrast = c("group", "dnFGFR_PN140_same", "dnFGFR_PN140_opposite"))
normed = counts(dds, normalized=TRUE)
resSig <- subset(res, padj < 0.05)
rs <- as.data.frame(resSig)
siggenes <- as.vector(rownames(rs))

# # Write out results
# setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\dn140\\")
# write.csv(as.data.frame(res), file="dn140_results.csv")
# 
# # Create MA Plot, alpha is FDR cutoff, for this experiment it is "0.05"
# setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\dn140\\")
# pdf("dn140_MAPlot.pdf")
# DESeq2::plotMA(res, alpha = 0.05, ylim=c(-3,3), cex=.4)
# abline(h=c(-1,1), col="dodgerblue", lwd=2)
# dev.off()

# # Write out normalized counts
# write.csv(normed4, file="all_normCounts.csv")
# 
# # Plot dispersions
# pdf("all_genotype-housing-time_dispersionPlot.pdf")
# plotDispEsts(dds4)
# dev.off()
```

# GSEA Results -----------------------------------------------------------------------------------------------------------------


```{r}
gsea_inflammatory <- read.table("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\GSEA\\HALLMARK_INFLAMMATORY_RESPONSE.tsv", sep = "\t", row.names = NULL, header = TRUE)

mininormed <- normed[noquote(toupper(as.character(rownames(normed)))) %in% noquote(unique(gsea_inflammatory$SYMBOL)),]
  mininormed <- as.matrix(mininormed)
  setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\GSEA\\")
  title = "Hallmark Inflammatory Response"
  png(filename = "inflammatory_response_w.png", width = 12, height = 12, units = "in", res = 800, bg= "white")
  heatmap(mininormed, Colv = NA, main=title)
  dev.off()
  
minisig140 <- select(as.data.frame(normed), contains("B"))
  minisig140 <- minisig140[noquote(toupper(as.character(rownames(minisig140)))) %in% noquote(unique(gsea_inflammatory$SYMBOL)),]
  minisig140 <- minisig140[noquote(rownames(minisig140)) %in% noquote(unique(siggenes)),]
  minisig140 <- as.matrix(minisig140)
  setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\GSEA\\")
  title = "Hallmark Inflammatory Response"
  png(filename = "inflammatory_response_sig.png", width = 12, height = 12, units = "in", res = 800, bg= "white")
  heatmap(minisig140, Colv = NA, main=title)
  dev.off()
  
gsea_inta <- read.table("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\GSEA\\HALLMARK_INTERFERON_ALPHA_RESPONSE.tsv", sep = "\t", row.names = NULL, header = TRUE)

mininormed <- normed[noquote(toupper(as.character(rownames(normed)))) %in% noquote(unique(gsea_inta$SYMBOL)),]
  mininormed <- as.matrix(mininormed)
  setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\GSEA\\")
  title = "Hallmark Interferon Alpha Response"
  png(filename = "interferon_alpha_response.png", width = 12, height = 12, units = "in", res = 800, bg= "white")
  heatmap(mininormed, Colv = NA, main=title)
  dev.off()
  
minisig140 <- select(as.data.frame(normed), contains("B"))
  minisig140 <- minisig140[noquote(toupper(as.character(rownames(minisig140)))) %in% noquote(unique(gsea_inta$SYMBOL)),]
  minisig140 <- minisig140[noquote(rownames(minisig140)) %in% noquote(unique(siggenes)),]
  minisig140 <- as.matrix(minisig140)
  setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\GSEA\\")
  title = "Hallmark Interferon Alpha Response"
  png(filename = "interferon_alpha_response_sig.png", width = 12, height = 12, units = "in", res = 800, bg= "white")
  heatmap(minisig140, Colv = NA, main=title)
  dev.off()

```


# GO Term Heatmaps ----------------------------------------------------------------------------------------------------------------


```{r}
go_procp <- read.table("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\Go_heatmaps\\positive_regulation_of_cytokine_production.txt", sep = "\t", row.names = NULL, header = TRUE, quote = "")

# normed140 <- as.data.frame(normed)
# normed140 <- select(normed140, contains("B"))
# normed140 <- log(normed140[!(apply(normed140, 1, function(y) any(y == 0))),])

mininormed <- normed[noquote(rownames(normed)) %in% noquote(unique(as.vector(go_procp$MGI.Gene.Marker.ID))),]
  mininormed <- as.matrix(mininormed)
  setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\Go_heatmaps\\")
  title = "Positive Regulation of Cytokine Production"
  png(filename = "positive_regulation_of_cytokine_production.png", width = 12, height = 12, units = "in", res = 800, bg= "NA")
  heatmap(mininormed, Colv = NA, main=title)
  dev.off()
  
minisig140 <- select(as.data.frame(normed), contains("B"))
  minisig140 <- minisig140[noquote(rownames(minisig140)) %in% noquote(unique(as.vector(go_procp$MGI.Gene.Marker.ID))),]
  minisig140 <- minisig140[noquote(rownames(minisig140)) %in% noquote(unique(siggenes)),]
  minisig140 <- as.matrix(minisig140)
  setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\Go_heatmaps\\")
  title = "Positive Regulation of Cytokine Production"
  png(filename = "positive_regulation_of_cytokine_production_sig.png", width = 12, height = 12, units = "in", res = 800, bg= "NA")
  heatmap(minisig140, Colv = NA, main=title)
  dev.off()

minisig60 <- select(as.data.frame(normed), contains("A"))
  minisig60 <- minisig60[noquote(rownames(minisig60)) %in% noquote(unique(as.vector(go_procp$MGI.Gene.Marker.ID))),]
  minisig60 <- minisig60[noquote(rownames(minisig60)) %in% noquote(unique(siggenes)),]
  minisig60 <- as.matrix(minisig60)
  setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\Go_heatmaps\\")
  title = "Positive Regulation of Cytokine Production"
  png(filename = "positive_regulation_of_cytokine_production.png_sig60", width = 12, height = 12, units = "in", res = 800, bg= "NA")
  heatmap(minisig60, Colv = NA, main=title)
  dev.off()

```


```{r}
go_iir <- read.table("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\Go_heatmaps\\innate_immune_response.txt", sep = "\t", row.names = NULL, header = TRUE, quote = "")

mininormed <- normed[noquote(rownames(normed)) %in% noquote(unique(as.vector(go_iir$MGI.Gene.Marker.ID))),]
  mininormed <- as.matrix(mininormed)
  setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\Go_heatmaps\\")
  title = "Innate Immune Response"
  png(filename = "innate_immune_response.png", width = 12, height = 12, units = "in", res = 800, bg= "NA")
  heatmap(mininormed, Colv = NA, main=title)
  dev.off()
  
minisig140 <- select(as.data.frame(normed), contains("B"))
  minisig140 <- minisig140[noquote(rownames(minisig140)) %in% noquote(unique(as.vector(go_iir$MGI.Gene.Marker.ID))),]
  minisig140 <- minisig140[noquote(rownames(minisig140)) %in% noquote(unique(siggenes)),]
  minisig140 <- as.matrix(minisig140)
  setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\Go_heatmaps\\")
  title = "Innate Immune Response"
  png(filename = "innate_immune_response_sig.png", width = 12, height = 12, units = "in", res = 800, bg= "NA")
  heatmap(minisig140, Colv = NA, main=title)
  dev.off()

minisig60 <- select(as.data.frame(normed), contains("A"))
  minisig60 <- minisig60[noquote(rownames(minisig60)) %in% noquote(unique(as.vector(go_iir$MGI.Gene.Marker.ID))),]
  minisig60 <- minisig60[noquote(rownames(minisig60)) %in% noquote(unique(siggenes)),]
  minisig60 <- as.matrix(minisig60)
  setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\Go_heatmaps\\")
  title = "Innate Immune Response"
  png(filename = "innate_immune_response_sig60.png", width = 12, height = 12, units = "in", res = 800, bg= "NA")
  heatmap(minisig60, Colv = NA, main=title)
  dev.off()

```


```{r}
go_ctrbs <- read.table("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\Go_heatmaps\\cellular_response_to_biotic_stimulus.txt", sep = "\t", row.names = NULL, header = TRUE, quote = "")

mininormed <- normed[noquote(rownames(normed)) %in% noquote(unique(as.vector(go_ctrbs$MGI.Gene.Marker.ID))),]
  mininormed <- as.matrix(mininormed)
  setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\Go_heatmaps\\")
  title = "Cellular Response to Biotic Stimulus"
  png(filename = "cellular_response_to_biotic_stimulus.png", width = 12, height = 12, units = "in", res = 800, bg= "NA")
  heatmap(mininormed, Colv = NA, main=title)
  dev.off()
  
minisig140 <- select(as.data.frame(normed), contains("B"))
  minisig140 <- minisig140[noquote(rownames(minisig140)) %in% noquote(unique(as.vector(go_crtbs$MGI.Gene.Marker.ID))),]
  minisig140 <- minisig140[noquote(rownames(minisig140)) %in% noquote(unique(siggenes)),]
  minisig140 <- as.matrix(minisig140)
  setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\Go_heatmaps\\")
  title = "Cellular Response to Biotic Stimulus"
  png(filename = "cellular_response_to_biotic_stimulus_sig.png", width = 12, height = 12, units = "in", res = 800, bg= "NA")
  heatmap(minisig140, Colv = NA, main=title)
  dev.off()

minisig60 <- select(as.data.frame(normed), contains("A"))
  minisig60 <- minisig60[noquote(rownames(minisig60)) %in% noquote(unique(as.vector(go_crtbs$MGI.Gene.Marker.ID))),]
  minisig60 <- minisig60[noquote(rownames(minisig60)) %in% noquote(unique(siggenes)),]
  minisig60 <- as.matrix(minisig60)
  setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\Go_heatmaps\\")
  title = "Cellular Response to Biotic Stimulus"
  png(filename = "cellular_response_to_biotic_stimulus_sig60.png", width = 12, height = 12, units = "in", res = 800, bg= "NA")
  heatmap(minisig60, Colv = NA, main=title)
  dev.off()

  # write.table(normcounts_group,paste0(outdir,outfilename,"/",'cluster',i,'.txt', sep=""), sep="\t",append = FALSE, quote = FALSE)}

```

```{r}
go_air <- read.table("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\Go_heatmaps\\adaptive_immune_response.txt", sep = "\t", row.names = NULL, header = TRUE, quote = "")

mininormed <- normed[noquote(rownames(normed)) %in% noquote(unique(as.vector(go_air$MGI.Gene.Marker.ID))),]
  mininormed <- as.matrix(mininormed)
  setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\Go_heatmaps\\")
  title = "Adaptive Immune Response"
  png(filename = "adaptive_immune_response.png", width = 12, height = 12, units = "in", res = 800, bg= "white")
  heatmap(mininormed, Colv = NA, main=title)
  dev.off()
  
minisig140 <- select(as.data.frame(normed), contains("B"))
  minisig140 <- minisig140[noquote(rownames(minisig140)) %in% noquote(unique(as.vector(go_air$MGI.Gene.Marker.ID))),]
  minisig140 <- minisig140[noquote(rownames(minisig140)) %in% noquote(unique(siggenes)),]
  minisig140 <- as.matrix(minisig140)
  setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\Go_heatmaps\\")
  title = "Adaptive Immune Response"
  png(filename = "adaptive_immune_response_sig.png", width = 12, height = 12, units = "in", res = 800, bg= "white")
  heatmap(minisig140, Colv = NA, main=title)
  dev.off()

minisig60 <- select(as.data.frame(normed), contains("A"))
  minisig60 <- minisig60[noquote(rownames(minisig60)) %in% noquote(unique(as.vector(go_air$MGI.Gene.Marker.ID))),]
  minisig60 <- minisig60[noquote(rownames(minisig60)) %in% noquote(unique(siggenes)),]
  minisig60 <- as.matrix(minisig60)
  setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\Go_heatmaps\\")
  title = "Adaptive Immune Response"
  png(filename = "adaptive_immune_response_sig60.png", width = 12, height = 12, units = "in", res = 800, bg= "NA")
  heatmap(minisig60, Colv = NA, main=title)
  dev.off()

  # write.table(normcounts_group,paste0(outdir,outfilename,"/",'cluster',i,'.txt', sep=""), sep="\t",append = FALSE, quote = FALSE)}

```
