---
title: "DESeq2"
output: html_document
---

```{r setup, include=FALSE}
require("knitr")
knitr::opts_chunk$set(echo = TRUE)
opts_knit$set(root.dir = "C:\\GitHub\\dnFGFR\\deseq_outputs\\")
library(dplyr)
library(DESeq2)
library(knitr)
library(ggplot2)
library(magrittr)
ALLOW_WGCNA_THREADS=8
```

#Load counts and sample info
```{r}
counts = read.table("all_counts.txt", header=TRUE, row.names=1, sep="\t")
coldata = read.table("all_coldata.txt", header=TRUE, sep="\t")
```

#remove length from counts table
```{r}
#counts['Length'] = NULL
```

# Filter data to show different cohorts

```{r}
require(dplyr)
PN140counts <- select(counts, contains("B"))
PN140coldata <- filter(coldata, time=="PN140")
PN140DNcounts <- select(PN140counts, contains("B2"))
PN140DNcoldata <- filter(PN140coldata, genotype =="dnFGFR")
PN140DNcoldata['time'] = NULL
```

#sort coldata so it matches counts table columns
```{r}
coldata_sort <-coldata[order(coldata$sample),]
```


#make deseq2 object
```{r}
dds = DESeqDataSetFromMatrix(countData = PN140DNcounts, colData = PN140DNcoldata, design= ~housing)
```

#transform counts for PCA plots
```{r}
rld <- rlog(dds, blind=FALSE)
vst <- vst(dds, blind=FALSE)
ntd <- normTransform(dds)
```

#plot SD of transformed data
```{r}
setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\pdfs\\")
library("vsn")
pdf("sdplot_shiftedLogtransformed.pdf")
meanSdPlot(assay(ntd))
dev.off()

pdf("sdplot_regLogtransformed.pdf")
meanSdPlot(assay(rld))
dev.off()

pdf("sdplot_varstabilizedtransformed.pdf")
meanSdPlot(assay(vst))
dev.off()
```

#check to see what is the dominant driver of gene expression differences
```{r}
setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\pdfs\\")
pdf("allsamples_genotype_PCA.pdf")
sampleDists <- dist(t(assay(vst)))
plotPCA(rld, intgroup=c("genotype"))
dev.off()

pdf("allsamples_treatment_PCA.pdf")
sampleDists <- dist(t(assay(vst)))
plotPCA(rld, intgroup=c("housing"))
dev.off()

pdf("allsamples_day_PCA.pdf")
sampleDists <- dist(t(assay(vst)))
#plotPCA(rld, intgroup=c("time"))
dev.off()
```


# PN140 -----------------------------------------------------------------------------------------------------------------------------------

# PN140 effect of housing on dnFGFR animals only
```{r}
#Designate Files
setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\PN140\\dn_housing\\")

dds = DESeqDataSetFromMatrix(countData = PN140DNcounts, colData = PN140DNcoldata, design = ~ housing)

#remove low count genes (anything less than 20 reads)
keep <- rowSums(counts(dds)) >= 20
dds <- dds[keep,]

#set comparison level, 
dds$housing <- relevel(dds$housing, ref = "same")

#perform normalization, dispersion estimate, and statistical tests
dds = DESeq(dds)

#plot dispersions
pdf("dn_housing_dispersionPlot.pdf")
plotDispEsts(dds )
dev.off()

#write out normalized counts
normed = counts(dds, normalized=TRUE)
write.csv(normed, file="dn_PN140_normCounts.csv")

#write out statistical results
res <- results(dds)
write.csv(as.data.frame(res), file="dn_PN140_housing_results.csv")

#plot fold changes with 2 fold blue line added, alpha is FDR cutoff, change to whatever cutoff you want
pdf("dn_PN140_housing_MAPlot.pdf")
DESeq2::plotMA(res, alpha = 0.1, ylim=c(-3,3), cex=.4)
abline(h=c(-1,1), col="dodgerblue", lwd=2)
dev.off()

```


# PN140 effect of housing
```{r}
#Designate Files
setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\PN140\\housing\\")

dds = DESeqDataSetFromMatrix(countData = PN140counts, colData = PN140coldata, design = ~ housing)

#remove low count genes (anything less than 20 reads)
keep <- rowSums(counts(dds)) >= 20
dds <- dds[keep,]

#set comparison level, 
dds$housing <- relevel(dds$housing, ref = "same")

#perform normalization, dispersion estimate, and statistical tests
dds = DESeq(dds)

#plot dispersions
pdf("housing_dispersionPlot.pdf")
plotDispEsts(dds )
dev.off()

#write out normalized counts
normed = counts(dds, normalized=TRUE)
write.csv(normed, file="PN140_normCounts.csv")

#write out statistical results
res <- results(dds)
write.csv(as.data.frame(res), file="PN140_housing_results.csv")

#plot fold changes with 2 fold blue line added, alpha is FDR cutoff, change to whatever cutoff you want
pdf("PN140_housing_MAPlot.pdf")
DESeq2::plotMA(res, alpha = 0.1, ylim=c(-3,3), cex=.4)
abline(h=c(-1,1), col="dodgerblue", lwd=2)
dev.off()

```


# PN140 effect of housing, controlling for genotype
```{r}
#Designate Files
setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\PN140\\housing-gen\\")

dds = DESeqDataSetFromMatrix(countData = PN140counts, colData = PN140coldata, design = ~genotype + housing)

#remove low count genes (anything less than 20 reads)
keep <- rowSums(counts(dds)) >= 20
dds <- dds[keep,]

#set comparison level, 
dds$housing <- relevel(dds$housing, ref = "same")

#perform normalization, dispersion estimate, and statistical tests
dds = DESeq(dds)

#plot dispersions
pdf("housing-gen_dispersionPlot.pdf")
plotDispEsts(dds )
dev.off()

#write out normalized counts
normed = counts(dds, normalized=TRUE)
write.csv(normed, file="PN140_normCounts.csv")

#write out statistical results
res <- results(dds)
write.csv(as.data.frame(res), file="PN140_housing-gen_results.csv")

#plot fold changes with 2 fold blue line added, alpha is FDR cutoff, change to whatever cutoff you want
pdf("PN140_housing-gen_MAPlot.pdf")
DESeq2::plotMA(res, alpha = 0.1, ylim=c(-3,3), cex=.4)
abline(h=c(-1,1), col="dodgerblue", lwd=2)
dev.off()

```


# PN140 effect of genotype, controlling for housing
```{r}
#Designate Files
setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\PN140\\genotype-housing\\")

dds = DESeqDataSetFromMatrix(countData = PN140counts, colData = PN140coldata, design = ~ housing + genotype)

#remove low count genes (anything less than 20 reads)
keep <- rowSums(counts(dds)) >= 20
dds <- dds[keep,]

#set comparison level, 
dds$housing <- relevel(dds$housing, ref = "same")

#perform normalization, dispersion estimate, and statistical tests
dds = DESeq(dds)

#plot dispersions
pdf("genotype-housing_dispersionPlot.pdf")
plotDispEsts(dds )
dev.off()

#write out normalized counts
normed = counts(dds, normalized=TRUE)
write.csv(normed, file="PN140_normCounts.csv") 

#write out statistical results
res <- results(dds)
write.csv(as.data.frame(res), file="PN140_genotype-housing_results.csv")

#plot fold changes with 2 fold blue line added, alpha is FDR cutoff, change to whatever cutoff you want
pdf("PN140_genotype-housing_MAPlot.pdf")
DESeq2::plotMA(res, alpha = 0.1, ylim=c(-3,3), cex=.4)
abline(h=c(-1,1), col="dodgerblue", lwd=2)
dev.off()

```




# All Samples ---------------------------------------------------------------------------------------------------------------------------------

# All samples effect of housing controlling for genotype and time
```{r}
#Designate Files
setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\all_samples\\housing-genotype-time\\")

dds = DESeqDataSetFromMatrix(countData = counts, colData = coldata, design = ~ genotype + time + housing)

#remove low count genes (anything less than 20 reads)
keep <- rowSums(counts(dds)) >= 20
dds <- dds[keep,]

#set comparison level, 
dds$housing <- relevel(dds$housing, ref = "same")
dds$genotype <- relevel(dds$genotype, ref = "ctrl")

#perform normalization, dispersion estimate, and statistical tests
dds = DESeq(dds)

#plot dispersions
pdf("housing-genotype-time_dispersionPlot.pdf")
plotDispEsts(dds )
dev.off()

#write out normalized counts
normed = counts(dds, normalized=TRUE)
write.csv(normed, file="all_normCounts.csv") 

#write out statistical results
res <- results(dds)
write.csv(as.data.frame(res), file="all_housing-genotype-time_results.csv")

#plot fold changes with 2 fold blue line added, alpha is FDR cutoff, change to whatever cutoff you want
pdf("all_housing-genotype-time_MAPlot.pdf")
DESeq2::plotMA(res, alpha = 0.1, ylim=c(-3,3), cex=.4)
abline(h=c(-1,1), col="dodgerblue", lwd=2)
dev.off()

```


# All samples effect of genotype controlling for housing and time
```{r}
#Designate Files
setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\all_samples\\genotype-time-housing\\")

dds = DESeqDataSetFromMatrix(countData = counts, colData = coldata, design = ~ time + housing + genotype)

#remove low count genes (anything less than 20 reads)
keep <- rowSums(counts(dds)) >= 20
dds <- dds[keep,]

#set comparison level, 
dds$housing <- relevel(dds$housing, ref = "same")
dds$genotype <- relevel(dds$genotype, ref = "ctrl")

#perform normalization, dispersion estimate, and statistical tests
dds = DESeq(dds)

#plot dispersions
pdf("genotype-time-housing_dispersionPlot.pdf")
plotDispEsts(dds )
dev.off()

#write out normalized counts
normed = counts(dds, normalized=TRUE)
write.csv(normed, file="all_normCounts.csv") 

#write out statistical results
res <- results(dds)
write.csv(as.data.frame(res), file="all_genotype-time-housing_results.csv")

#plot fold changes with 2 fold blue line added, alpha is FDR cutoff, change to whatever cutoff you want
pdf("all_genotype-time-housing_MAPlot.pdf")
DESeq2::plotMA(res, alpha = 0.1, ylim=c(-3,3), cex=.4)
abline(h=c(-1,1), col="dodgerblue", lwd=2)
dev.off()

```



# All Samples, effect of time controlling for genotype and housing
```{r}
#Designate Files
setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\all_samples\\time-housing-genotype\\")

dds = DESeqDataSetFromMatrix(countData = counts, colData = coldata, design = ~ housing + genotype + time)

#remove low count genes (anything less than 20 reads)
keep <- rowSums(counts(dds)) >= 20
dds <- dds[keep,]

#set comparison level, 
dds$housing <- relevel(dds$housing, ref = "same")
dds$genotype <- relevel(dds$genotype, ref = "ctrl")

#perform normalization, dispersion estimate, and statistical tests
dds = DESeq(dds)

#plot dispersions
pdf("time-housing-genotype_dispersionPlot.pdf")
plotDispEsts(dds )
dev.off()

#write out normalized counts
normed = counts(dds, normalized=TRUE)
write.csv(normed, file="all_normCounts.csv") 

#write out statistical results
res <- results(dds)
write.csv(as.data.frame(res), file="all_time-housing-genotype_results.csv")

#plot fold changes with 2 fold blue line added, alpha is FDR cutoff, change to whatever cutoff you want
pdf("all_time-housing-genotype_MAPlot.pdf")
DESeq2::plotMA(res, alpha = 0.1, ylim=c(-3,3), cex=.4)
abline(h=c(-1,1), col="dodgerblue", lwd=2)
dev.off()

```














# OLD SCRIPT: PN140 dnFGFR effect of housing
```{r}
setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\test\\")
#Designate Files
counts = read.table("PN140dnFGFRcounts.txt", header=TRUE, row.names=1, sep="\t")
coldata = read.table("PN140dnFGFRcoldata.txt", header=TRUE, row.names=1, sep="\t")

#set dds object, define design
dds = DESeqDataSetFromMatrix(countData = counts, colData = coldata, design = ~housing)

#remove low count genes (anything less than 20 reads)
keep <- rowSums(counts(dds)) >= 20
dds <- dds[keep,]

#set comparison level, 
dds$housing <- relevel(dds$housing, ref = "same")
# dds$genotype <- relevel(dds$genotype, ref = "ctrl")

#perform normalization, dispersion estimate, and statistical tests
dds = DESeq(dds)

#plot dispersions
pdf("PN140dnFGFRdispersionPlot.pdf")
plotDispEsts(dds )
dev.off()

#write out normalized counts
normed = counts(dds, normalized=TRUE)
write.csv(normed, file="PN140dnFGFRNormCounts.csv")

#write out statistical results
res <- results(dds)
write.csv(as.data.frame(res), file="PN140dnFGFR_results.csv")

#plot fold changes with 2 fold blue line added, alpha is FDR cutoff, change to whatever cutoff you want
pdf("PN140dnFGFR_MAPlot.pdf")
DESeq2::plotMA(res, alpha = 0.1, ylim=c(-3,3), cex=.4)
abline(h=c(-1,1), col="dodgerblue", lwd=2)
dev.off()

```



#Checking Results PN140 (WIP)
```{r}
goi <- "ENSMUSG00000037461"
goi.count <- as.data.frame(normed[goi,])
names(goi.count) <- goi
t.goi.count <- as.data.frame(t(goi.count))


COS140 <- dplyr::filter(goi.count, contains("B110"))
CSS140 <- dplyr::select(goi.count, contains("B111"))
DOS140 <- dplyr::select(goi.count, contains("B210"))
DSS140 <- dplyr::select(goi.count, contains("B211"))
goi.sorted <- data.frame(COS140, CSS140, DOS140, DSS140)
names(goi.sorted) <- data.frame("C-SS", "C-OS", "D-SS", "D-OS")

```



















