---
title: "DESeq2.v2"
author: "Tyler Akonom"
date: "12/4/2020"
output: html_document
---

```{r setup, include=FALSE}
require("knitr")
knitr::opts_chunk$set(echo = TRUE)
opts_knit$set(root.dir = "C:\\GitHub\\dnFGFR\\deseq_outputs\\")
library(dplyr)
library(DESeq2)
library(knitr)
library(ggplot2)
library(magrittr)
library(tidyr)
library(tibble)
library(DEGreport)
library(vsn)
```


# Load counts and metadata
```{r}
counts = read.table("refcounts.txt", header=TRUE, row.names=1, sep="\t")
coldata = read.table("coldata.txt", header=TRUE, sep="\t")
coldata$group<-paste(coldata$genotype, coldata$time, coldata$housing, sep="_")
coldata$timehousing <- paste(coldata$time, coldata$housing, sep="_")
coldata$housing <- factor(coldata$housing, levels = c("same","opposite"))
coldata$genotype <- factor(coldata$genotype, levels = c("ctrl", "dnFGFR"))
coldata$time <- factor(coldata$time, levels = c("PN60",  "PN140"))
coldata$group <- factor(coldata$group)
coldata$timehousing <- factor(coldata$timehousing)
coldata$parent <- factor(coldata$parent)
coldata$dod <- factor(coldata$dod)

#remove length from counts table and sort properly
counts['Length'] = NULL
counts <- counts[,coldata$sampleName]
```


# Filter data to show different cohorts
```{r}
# require(dplyr)
# PN140counts <- select(counts, contains("B"))
# PN140coldata <- filter(coldata, time=="PN140")
# PN140DNcounts <- select(PN140counts, contains("B2"))
# PN140DNcoldata <- filter(PN140coldata, genotype =="dnFGFR")
# PN140DNcoldata['time'] = NULL
# 
# PN140Ctrlcounts <- select(PN140counts, contains("B1"))
# PN140Ctrlcoldata <- filter(PN140coldata, genotype =="ctrl")
# PN140Ctrlcoldata['time'] = NULL
# 
# PN140SScounts <- select(PN140counts, contains(c("B1116","B1117","B1118","B1119","B1120","B2111","B2112","B2113","B2114","B2115")))
# PN140SScoldata <- filter(PN140coldata, housing =="same")
# PN140SScoldata['time'] = NULL
# 
# PN60counts <- select(counts, contains("A"))
# PN60coldata <- filter(coldata, time=="PN60")
# PN60DNcounts <- select(PN60counts, contains("A2"))
# PN60DNcoldata <- filter(PN60coldata, genotype =="dnFGFR")
# PN60DNcoldata['time'] = NULL

```

# Checking to see if parent cage or date of collection drives expression
```{r}
# dds = DESeqDataSetFromMatrix(countData = counts, colData = coldata, design = ~dod)
# ntd <- normTransform(dds)
# meanSdPlot(assay(ntd))
# sampleDists <- dist(t(assay(ntd)))
# pdf("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\figures\\PCA_DOD.pdf")
# plotPCA(ntd, intgroup=c("dod"))
# dev.off()
```

```{r}
# dds = DESeqDataSetFromMatrix(countData = counts, colData = coldata, design = ~parent)
# ntd <- normTransform(dds)
# meanSdPlot(assay(ntd))
# sampleDists <- dist(t(assay(ntd)))
# pdf("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\figures\\PCA_Parent.pdf")
# plotPCA(ntd, intgroup=c("parent"))
# dev.off()
```

# Creating a DESeq object and check for differential expression -----------------------------------
# PN140 -------------------------------------------------------------------------------------------

```{r}
setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\")

# Create DESeq object
dds = DESeqDataSetFromMatrix(countData = counts, colData = coldata, design = ~genotype  + timehousing)

# Run DESeq on the DESeq object
dds = DESeq(dds)
res <- results(dds, contrast = c("timehousing", "PN140_same", "PN140_opposite"))
normed = counts(dds, normalized=TRUE)
resSig <- subset(res, padj < 0.05)
rs <- as.data.frame(resSig)
siggenes <- as.vector(rownames(rs))

# Creating a PDF of each gene identified above
setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\figures\\")
for (genename in siggenes){
  n <- paste(genename,"pdf", sep = ".")
  pdf(file = n, onefile = FALSE, title = genename)
  plotCounts(dds,gene=genename,intgroup="timehousing")
  dev.off()
}

setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\timehousing\\housing\\")
# # Write out results
# write.csv(as.data.frame(resallg), file="all_genotype-housing-time_results.csv")
# 
# # Create MA Plot, alpha is FDR cutoff, for this experiment it is "0.05"
# pdf("all_genotype-housing-time_MAPlot.pdf")
# DESeq2::plotMA(resallg, alpha = 0.05, ylim=c(-3,3), cex=.4)
# abline(h=c(-1,1), col="dodgerblue", lwd=2)
# dev.off()
# 
# # Write out normalized counts
# write.csv(normed4, file="all_normCounts.csv")
# 
# # Plot dispersions
# pdf("all_genotype-housing-time_dispersionPlot.pdf")
# plotDispEsts(dds4)
# dev.off()
```

```{r}
gotermgenes <- read.table("C:\\GitHub\\dnFGFR\\Go_inflammatory_response.txt", sep = "\t", row.names = NULL, header = TRUE)

mininormed <- normed[noquote(rownames(normed)) %in% noquote(unique(as.vector(gotermgenes$MGI.Gene.Marker.ID))),]
  mininormed <- as.matrix(mininormed)
  title = "neatgenes"
  # jpeg(paste0(outdir,outfilename,"/",'cluster',i,'.jpg', sep=""))
  heatmap(mininormed,Colv = NA, main=title)
  # dev.off()
  # write.table(normcounts_group,paste0(outdir,outfilename,"/",'cluster',i,'.txt', sep=""), sep="\t",append = FALSE, quote = FALSE)}

```
