---
title: "DESeq2"
output: html_document
---

```{r setup, include=FALSE}
require("knitr")
knitr::opts_chunk$set(echo = TRUE)
opts_knit$set(root.dir = "C:\\GitHub\\dnFGFR\\deseq_outputs\\")
library(dplyr)
library(DESeq2)
library(knitr)
library(ggplot2)
library(magrittr)
library(tidyr)
library(tibble)
library(DEGreport)
```

#Load counts and sample info
```{r}
counts = read.table("refcounts.txt", header=TRUE, row.names=1, sep="\t")
coldata = read.table("coldata.txt", header=TRUE, sep="\t")
```

#remove length from counts table
```{r}
counts['Length'] = NULL
counts <- counts[,coldata$sampleName]
```

#sort coldata so it matches counts table columns
```{r}
#coldata_s <-coldata[order(colnames(counts)),]
#rownames(coldata) = coldata$sampleName
#coldata['sampleName'] = NULL
#rownames(coldata_s) = coldata_s$sampleName
#coldata_s['sampleName'] = NULL
#coldata <- coldata_s
```

# Filter data to show different cohorts

```{r}
require(dplyr)
PN140counts <- select(counts, contains("B"))
PN140coldata <- filter(coldata, time=="PN140")
PN140DNcounts <- select(PN140counts, contains("B2"))
PN140DNcoldata <- filter(PN140coldata, genotype =="dnFGFR")
PN140DNcoldata['time'] = NULL

```



#make deseq2 object
```{r}
dds = DESeqDataSetFromMatrix(countData = counts, colData = coldata, design= ~ time + genotype + housing)
```

#transform counts for PCA plots
```{r}
rld <- rlog(dds, blind=TRUE)
vst <- vst(dds, blind=FALSE)
ntd <- normTransform(dds)
```

#plot SD of transformed data
```{r}
setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\pdfs\\")
library("vsn")
pdf("sdplot_shiftedLogtransformed.pdf")
meanSdPlot(assay(ntd))
dev.off()

pdf("sdplot_regLogtransformed.pdf")
meanSdPlot(assay(rld))
dev.off()

pdf("sdplot_varstabilizedtransformed.pdf")
meanSdPlot(assay(vst))
dev.off()
```

#check to see what is the dominant driver of gene expression differences
```{r}

setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\pdfs\\")

pdf("genotype_PCA.pdf")
sampleDists <- dist(t(assay(ntd)))
plotPCA(ntd, intgroup=c("genotype"))
dev.off()

pdf("treatment_PCA.pdf")
sampleDists <- dist(t(assay(ntd)))
plotPCA(ntd, intgroup=c("housing"))
dev.off()

pdf("day_PCA.pdf")
sampleDists <- dist(t(assay(ntd)))
plotPCA(ntd, intgroup=c("time"))
dev.off()
```

# Visualizing Neat Groups -----------------------------------------------------------------------------------------------------------------------------------


```{r}
dds_lrt <- DESeq(dds, test="LRT", reduced = ~ 1)
normcounts <- as.data.frame(counts(dds_lrt, normalize=TRUE))
normcountslong <- normcounts %>% gather(key = "sample", value = "signal")
ggplot(normcountslong, aes(x = sample, y = signal))+ geom_violin(trim = FALSE) + theme(axis.text.x = element_text(angle = 90))+scale_y_continuous(trans='log2')
genename="Tnf"
normcounts[genename,]
plotCounts(dds_lrt, intgroup="housing",genename)
res_LRT <- results(dds_lrt)
# Subset the LRT results to return genes with padj < 0.05
padj.cutoff <-0.01
sig_res_LRT <- res_LRT %>%
  data.frame() %>%
  rownames_to_column(var="gene") %>% 
  as_tibble() %>% 
  filter(padj < padj.cutoff)

# Get sig gene lists
sigLRT_genes <- sig_res_LRT %>% 
  pull(gene)

length(sigLRT_genes)
# Subset results for faster cluster finding (for classroom demo purposes)

clustering_sig_genes <- sig_res_LRT %>%
  arrange(padj) %>%
  head(n=1000)

#when you do alll genes you should do this instead
#clustering_sig_genes <- sig_res_LRT

# Obtain rlog values for those significant 
rld <- rlog(dds_lrt, blind=TRUE)
rld_mat <- assay(rld)
cluster_rlog <- rld_mat[clustering_sig_genes$gene, ]

#make meta
meta <-as.data.frame(cbind(paste(dds_lrt$time),paste(dds_lrt$housing),paste(dds_lrt$genotype),paste(dds_lrt$sampleName)))
colnames(meta)<-c("time","housing","genotype","sampleName")
row.names(meta)<-meta$sampleName
meta$time <- factor(meta$time, levels = c("PN60", "PN140"))
meta$housingtime <- factor(paste0(meta$time,"_", meta$housing,sep = ""), levels = c("PN60_same", "PN140_same", "PN60_opposite", "PN140_opposite"))


# jpeg(paste0(outdir,outfilename,"/",'clusters.jpg', sep=""))
clusters <- degPatterns(cluster_rlog, metadata = meta, time = "housingtime", col=NULL)
# dev.off()

#get more info on clusters
cluster_groups <- clusters$df
for (i in unique(cluster_groups$cluster)){
  group <- clusters$df %>% filter(cluster == i)
  normcounts_group <- unnorm_counts[rownames(unnorm_counts) %in% noquote(group$genes),]
  normcounts_group <- as.matrix(normcounts_group)
  title = paste0("cluster", i, sep=" ")
  jpeg(paste0(outdir,outfilename,"/",'cluster',i,'.jpg', sep=""))
  heatmap(normcounts_group,Colv = NA, main=title)
  dev.off()
  write.table(normcounts_group,paste0(outdir,outfilename,"/",'cluster',i,'.txt', sep=""), sep="\t",append = FALSE, quote = FALSE)}
```








# PN140 -----------------------------------------------------------------------------------------------------------------------------------

# PN140 effect of housing on dnFGFR animals only
```{r}
#Designate Files
setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\PN140\\dn_housing\\")

dds = DESeqDataSetFromMatrix(countData = PN140DNcounts, colData = PN140DNcoldata, design = ~ housing)

#remove low count genes (anything less than 20 reads)
#keep <- rowSums(counts(dds)) >= 20
#dds <- dds[keep,]

#set comparison level, 
dds$housing <- relevel(dds$housing, ref = "same")

#perform normalization, dispersion estimate, and statistical tests
dds = DESeq(dds)

#plot dispersions
pdf("dn_housing_dispersionPlot.pdf")
plotDispEsts(dds)
dev.off()

#write out normalized counts
normed = counts(dds, normalized=TRUE)
write.csv(normed, file="dn_PN140_normCounts.csv")

#write out statistical results
res <- results(dds)
write.csv(as.data.frame(res), file="dn_PN140_housing_results.csv")

#plot fold changes with 2 fold blue line added, alpha is FDR cutoff, change to whatever cutoff you want
pdf("dn_PN140_housing_MAPlot.pdf")
DESeq2::plotMA(res, alpha = 0.1, ylim=c(-3,3), cex=.4)
abline(h=c(-1,1), col="dodgerblue", lwd=2)
dev.off()

plotCounts(dds,gene="Tnf",intgroup="housing")

```


# PN140 effect of housing
```{r}
#Designate Files
setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\PN140\\housing\\")

dds = DESeqDataSetFromMatrix(countData = PN140counts, colData = PN140coldata, design = ~ housing)

#remove low count genes (anything less than 20 reads)
#keep <- rowSums(counts(dds)) >= 20
#dds <- dds[keep,]

#set comparison level, 
dds$housing <- relevel(dds$housing, ref = "same")

#perform normalization, dispersion estimate, and statistical tests
dds = DESeq(dds)

#plot dispersions
pdf("housing_dispersionPlot.pdf")
plotDispEsts(dds )
dev.off()

#write out normalized counts
normed = counts(dds, normalized=TRUE)
write.csv(normed, file="PN140_normCounts.csv")

#write out statistical results
res <- results(dds)
write.csv(as.data.frame(res), file="PN140_housing_results.csv")

#plot fold changes with 2 fold blue line added, alpha is FDR cutoff, change to whatever cutoff you want
pdf("PN140_housing_MAPlot.pdf")
DESeq2::plotMA(res, alpha = 0.1, ylim=c(-3,3), cex=.4)
abline(h=c(-1,1), col="dodgerblue", lwd=2)
dev.off()

```


# PN140 effect of housing, controlling for genotype
```{r}
#Designate Files
setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\PN140\\housing-gen\\")

dds = DESeqDataSetFromMatrix(countData = PN140counts, colData = PN140coldata, design = ~genotype + housing)

#remove low count genes (anything less than 20 reads)
#keep <- rowSums(counts(dds)) >= 20
#dds <- dds[keep,]

#set comparison level, 
dds$housing <- relevel(dds$housing, ref = "same")

#perform normalization, dispersion estimate, and statistical tests
dds = DESeq(dds)

#plot dispersions
pdf("housing-gen_dispersionPlot.pdf")
plotDispEsts(dds )
dev.off()

#write out normalized counts
normed = counts(dds, normalized=TRUE)
write.csv(normed, file="PN140_normCounts.csv")

#write out statistical results
res <- results(dds)
write.csv(as.data.frame(res), file="PN140_housing-gen_results.csv")

#plot fold changes with 2 fold blue line added, alpha is FDR cutoff, change to whatever cutoff you want
pdf("PN140_housing-gen_MAPlot.pdf")
DESeq2::plotMA(res, alpha = 0.1, ylim=c(-3,3), cex=.4)
abline(h=c(-1,1), col="dodgerblue", lwd=2)
dev.off()

```


# PN140 effect of genotype, controlling for housing
```{r}
#Designate Files
setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\PN140\\genotype-housing\\")

dds = DESeqDataSetFromMatrix(countData = PN140counts, colData = PN140coldata, design = ~ housing + genotype)

#remove low count genes (anything less than 20 reads)
#keep <- rowSums(counts(dds)) >= 20
#dds <- dds[keep,]

#set comparison level, 
dds$housing <- relevel(dds$housing, ref = "same")

#perform normalization, dispersion estimate, and statistical tests
dds = DESeq(dds)

#plot dispersions
pdf("genotype-housing_dispersionPlot.pdf")
plotDispEsts(dds )
dev.off()

#write out normalized counts
normed = counts(dds, normalized=TRUE)
write.csv(normed, file="PN140_normCounts.csv") 

#write out statistical results
res <- results(dds)
write.csv(as.data.frame(res), file="PN140_genotype-housing_results.csv")

#plot fold changes with 2 fold blue line added, alpha is FDR cutoff, change to whatever cutoff you want
pdf("PN140_genotype-housing_MAPlot.pdf")
DESeq2::plotMA(res, alpha = 0.1, ylim=c(-3,3), cex=.4)
abline(h=c(-1,1), col="dodgerblue", lwd=2)
dev.off()

```




# All Samples ---------------------------------------------------------------------------------------------------------------------------------

# All samples effect of housing controlling for genotype and time
```{r}
#Designate Files
setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\all_samples\\housing-genotype-time\\")

dds = DESeqDataSetFromMatrix(countData = counts, colData = coldata, design = ~ genotype + time + housing)

#remove low count genes (anything less than 20 reads)
#keep <- rowSums(counts(dds)) >= 20
#dds <- dds[keep,]

#set comparison level, 
dds$housing <- relevel(dds$housing, ref = "same")
dds$genotype <- relevel(dds$genotype, ref = "ctrl")

#perform normalization, dispersion estimate, and statistical tests
dds = DESeq(dds)

#plot dispersions
pdf("housing-genotype-time_dispersionPlot.pdf")
plotDispEsts(dds )
dev.off()

#write out normalized counts
normed = counts(dds, normalized=TRUE)
write.csv(normed, file="all_normCounts.csv") 

#write out statistical results
res <- results(dds)
write.csv(as.data.frame(res), file="all_housing-genotype-time_results.csv")

#plot fold changes with 2 fold blue line added, alpha is FDR cutoff, change to whatever cutoff you want
pdf("all_housing-genotype-time_MAPlot.pdf")
DESeq2::plotMA(res, alpha = 0.1, ylim=c(-3,3), cex=.4)
abline(h=c(-1,1), col="dodgerblue", lwd=2)
dev.off()

```


# All samples effect of genotype controlling for housing and time
```{r}
#Designate Files
setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\all_samples\\genotype-time-housing\\")

dds = DESeqDataSetFromMatrix(countData = counts, colData = coldata, design = ~ time + housing + genotype)

#remove low count genes (anything less than 20 reads)
#keep <- rowSums(counts(dds)) >= 20
#dds <- dds[keep,]

#set comparison level, 
dds$housing <- relevel(dds$housing, ref = "same")
dds$genotype <- relevel(dds$genotype, ref = "ctrl")

#perform normalization, dispersion estimate, and statistical tests
dds = DESeq(dds)

#plot dispersions 
pdf("genotype-time-housing_dispersionPlot.pdf")
plotDispEsts(dds )
dev.off()

#write out normalized counts
normed = counts(dds, normalized=TRUE)
write.csv(normed, file="all_normCounts.csv") 

#write out statistical results
res <- results(dds)
write.csv(as.data.frame(res), file="all_genotype-time-housing_results.csv")

#plot fold changes with 2 fold blue line added, alpha is FDR cutoff, change to whatever cutoff you want
pdf("all_genotype-time-housing_MAPlot.pdf")
DESeq2::plotMA(res, alpha = 0.1, ylim=c(-3,3), cex=.4)
abline(h=c(-1,1), col="dodgerblue", lwd=2)
dev.off()

```



# All Samples, effect of time controlling for genotype and housing
```{r}
#Designate Files
setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\all_samples\\time-housing-genotype\\")

dds = DESeqDataSetFromMatrix(countData = counts, colData = coldata, design = ~ housing + genotype + time)

#remove low count genes (anything less than 20 reads)
keep <- rowSums(counts(dds)) >= 20
dds <- dds[keep,]

#set comparison level, 
dds$housing <- relevel(dds$housing, ref = "same")
dds$genotype <- relevel(dds$genotype, ref = "ctrl")

#perform normalization, dispersion estimate, and statistical tests
dds = DESeq(dds)

#plot dispersions
pdf("time-housing-genotype_dispersionPlot.pdf")
plotDispEsts(dds )
dev.off()

#write out normalized counts
normed = counts(dds, normalized=TRUE)
write.csv(normed, file="all_normCounts.csv") 

#write out statistical results
res <- results(dds)
write.csv(as.data.frame(res), file="all_time-housing-genotype_results.csv")

#plot fold changes with 2 fold blue line added, alpha is FDR cutoff, change to whatever cutoff you want
pdf("all_time-housing-genotype_MAPlot.pdf")
DESeq2::plotMA(res, alpha = 0.1, ylim=c(-3,3), cex=.4)
abline(h=c(-1,1), col="dodgerblue", lwd=2)
dev.off()

```














# OLD SCRIPT: PN140 dnFGFR effect of housing
```{r}
setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\test\\")
#Designate Files
counts = read.table("PN140dnFGFRcounts.txt", header=TRUE, row.names=1, sep="\t")
coldata = read.table("PN140dnFGFRcoldata.txt", header=TRUE, row.names=1, sep="\t")

#set dds object, define design
dds = DESeqDataSetFromMatrix(countData = counts, colData = coldata, design = ~housing)

#remove low count genes (anything less than 20 reads)
keep <- rowSums(counts(dds)) >= 20
dds <- dds[keep,]

#set comparison level, 
dds$housing <- relevel(dds$housing, ref = "same")
# dds$genotype <- relevel(dds$genotype, ref = "ctrl")

#perform normalization, dispersion estimate, and statistical tests
dds = DESeq(dds)

#plot dispersions
pdf("PN140dnFGFRdispersionPlot.pdf")
plotDispEsts(dds )
dev.off()

#write out normalized counts
normed = counts(dds, normalized=TRUE)
write.csv(normed, file="PN140dnFGFRNormCounts.csv")

#write out statistical results
res <- results(dds)
write.csv(as.data.frame(res), file="PN140dnFGFR_results.csv")

#plot fold changes with 2 fold blue line added, alpha is FDR cutoff, change to whatever cutoff you want
pdf("PN140dnFGFR_MAPlot.pdf")
DESeq2::plotMA(res, alpha = 0.1, ylim=c(-3,3), cex=.4)
abline(h=c(-1,1), col="dodgerblue", lwd=2)
dev.off()

```



#Checking Results PN140 (WIP)
```{r}
goi <- "ENSMUSG00000037461"
goi.count <- as.data.frame(normed[goi,])
names(goi.count) <- goi
t.goi.count <- as.data.frame(t(goi.count))


COS140 <- dplyr::filter(goi.count, contains("B110"))
CSS140 <- dplyr::select(goi.count, contains("B111"))
DOS140 <- dplyr::select(goi.count, contains("B210"))
DSS140 <- dplyr::select(goi.count, contains("B211"))
goi.sorted <- data.frame(COS140, CSS140, DOS140, DSS140)
names(goi.sorted) <- data.frame("C-SS", "C-OS", "D-SS", "D-OS")

```



















