---
title: "DESeq2.v2"
author: "Tyler Akonom"
date: "12/4/2020"
output: html_document
---

```{r setup, include=FALSE}
require("knitr")
knitr::opts_chunk$set(echo = TRUE)
opts_knit$set(root.dir = "C:\\GitHub\\dnFGFR\\deseq_outputs\\")
library(dplyr)
library(DESeq2)
library(knitr)
library(ggplot2)
library(magrittr)
library(tidyr)
library(tibble)
library(DEGreport)
```


# Load counts and metadata
```{r}
counts = read.table("refcounts.txt", header=TRUE, row.names=1, sep="\t")
coldata = read.table("coldata.txt", header=TRUE, sep="\t")
```

#remove length from counts table and sort properly
```{r}
counts['Length'] = NULL
counts <- counts[,coldata$sampleName]
```

# Filter data to show different cohorts
```{r}
require(dplyr)
PN140counts <- select(counts, contains("B"))
PN140coldata <- filter(coldata, time=="PN140")
PN140DNcounts <- select(PN140counts, contains("B2"))
PN140DNcoldata <- filter(PN140coldata, genotype =="dnFGFR")
PN140DNcoldata['time'] = NULL

PN140Ctrlcounts <- select(PN140counts, contains("B1"))
PN140Ctrlcoldata <- filter(PN140coldata, genotype =="ctrl")
PN140Ctrlcoldata['time'] = NULL

PN140SScounts <- select(PN140counts, contains(c("B1116","B1117","B1118","B1119","B1120","B2111","B2112","B2113","B2114","B2115")))
PN140SScoldata <- filter(PN140coldata, housing =="same")
PN140SScoldata['time'] = NULL

PN60counts <- select(counts, contains("A"))
PN60coldata <- filter(coldata, time=="PN60")
PN60DNcounts <- select(PN60counts, contains("A2"))
PN60DNcoldata <- filter(PN60coldata, genotype =="dnFGFR")
PN60DNcoldata['time'] = NULL

```

# Creating a DESeq object and check for differential expression -----------------------------------
# PN140 -------------------------------------------------------------------------------------------

# PN140 dnFGFR animals only (dds1) (effect of housing on dnFGFR animals)
```{r}
setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\PN140\\dn_housing\\")

# Create DESeq object
dds1 = DESeqDataSetFromMatrix(countData = PN140DNcounts, colData = PN140DNcoldata, design = ~housing)

# Set "reference" levels
dds1$housing <- relevel(dds1$housing, ref = "same")

# Run DESeq on the DESeq object
dds1 = DESeq(dds1)
res140DN <- results(dds1)
normed1 = counts(dds1, normalized=TRUE)

# Check individual genes
plotCounts(dds1,gene="Gpnmb",intgroup="housing")
# 
# # Write out results
# write.csv(as.data.frame(res140DN), file="PN140DN_housing_results.csv")

# # Create MA Plot, alpha is FDR cutoff, for this experiment it is "0.05"
# pdf("PN140DN_housing_MAPlot.pdf")
# DESeq2::plotMA(res140DN, alpha = 0.05, ylim=c(-3,3), cex=.4)
# abline(h=c(-1,1), col="dodgerblue", lwd=2)
# dev.off()

# # Write out normalized counts
# write.csv(normed1, file="PN140DN_normCounts.csv")

# # Plot dispersions
# pdf("PN140DN_housing_dispersionPlot.pdf")
# plotDispEsts(dds1)
# dev.off()
# 
# normed1[rownames(normed1)=="Gpnmb",]
# barplot(normed1[rownames(normed1)=="Gpnmb",])

```

# PN140 Control animals only (dds9) (effect of housing on control animals)
```{r}
setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\PN140\\ctrl_housing\\")

# Create DESeq object
dds9 = DESeqDataSetFromMatrix(countData = PN140Ctrlcounts, colData = PN140Ctrlcoldata, design = ~housing)

# Set "reference" levels
dds9$housing <- relevel(dds9$housing, ref = "same")

# Run DESeq on the DESeq object
dds9 = DESeq(dds9)
res140Ctrl <- results(dds9)
normed1 = counts(dds9, normalized=TRUE)

# # Check individual genes
# plotCounts(dds9,gene="Ccl5",intgroup="housing")
# 
# # Write out results
# write.csv(as.data.frame(res140Ctrl), file="PN140Ctrl_housing_results.csv")
# 
# # Create MA Plot, alpha is FDR cutoff, for this experiment it is "0.05"
# pdf("PN140Ctrl_housing_MAPlot.pdf")
# DESeq2::plotMA(res140Ctrl, alpha = 0.05, ylim=c(-3,3), cex=.4)
# abline(h=c(-1,1), col="dodgerblue", lwd=2)
# dev.off()
# 
# # Write out normalized counts
# write.csv(normed1, file="PN140Ctrl_normCounts.csv")
# 
# # Plot dispersions
# pdf("PN140Ctrl_housing_dispersionPlot.pdf")
# plotDispEsts(dds9)
# dev.off()
# 
# normed1[rownames(normed1)=="Gpnmb",]
# barplot(normed1[rownames(normed1)=="Gpnmb",])

```

# PN140 effect of housing, removing genotype effect (dds2)
```{r}
setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\PN140\\housing-gen\\")

# Create DESeq object
dds2 = DESeqDataSetFromMatrix(countData = PN140counts, colData = PN140coldata, design = ~genotype + housing)

# Set "reference" levels
dds2$housing <- relevel(dds2$housing, ref = "same")
dds2$genotype <- relevel(dds2$genotype, ref = "ctrl")

# Run DESeq on the DESeq object
dds2 = DESeq(dds2)
res140h <- results(dds2)
normed2 = counts(dds2, normalized=TRUE)

# # Check individual genes
# plotCounts(dds2,gene="Gpnmb",intgroup="housing")

# # Write out results
# write.csv(as.data.frame(res140h), file="PN140_housing-gen_results.csv")
# 
# # Create MA Plot, alpha is FDR cutoff, for this experiment it is "0.05"
# pdf("PN140_housing-gen_MAPlot.pdf")
# DESeq2::plotMA(res140h, alpha = 0.05, ylim=c(-3,3), cex=.4)
# abline(h=c(-1,1), col="dodgerblue", lwd=2)
# dev.off()
# 
# # Write out normalized counts
# write.csv(normed2, file="PN140_normCounts.csv")
# 
# # Plot dispersions
# pdf("PN140_housing-gen_dispersionPlot.pdf")
# plotDispEsts(dds2)
# dev.off()

```

# PN140 effect of genotype, removing housing effect (dds3)
```{r}
setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\PN140\\genotype-housing\\")

# Create DESeq object
dds3 = DESeqDataSetFromMatrix(countData = PN140counts, colData = PN140coldata, design = ~housing + genotype)

# Set "reference" levels
dds3$housing <- relevel(dds3$housing, ref = "same")
dds3$genotype <- relevel(dds3$genotype, ref = "ctrl")

# Run DESeq on the DESeq object
dds3 = DESeq(dds3)
res140g <- results(dds3)
normed3 = counts(dds3, normalized=TRUE)

# # Check individual genes
# plotCounts(dds3,gene="Rnf122",intgroup="genotype")
# 
# # Write out results
# write.csv(as.data.frame(res140g), file="PN140_genotype-housing_results.csv")
# 
# # Create MA Plot, alpha is FDR cutoff, for this experiment it is "0.05"
# pdf("PN140_genotype-housing_MAPlot.pdf")
# DESeq2::plotMA(res140g, alpha = 0.05, ylim=c(-3,3), cex=.4)
# abline(h=c(-1,1), col="dodgerblue", lwd=2)
# dev.off()
# 
# # Write out normalized counts
# write.csv(normed3, file="PN140_normCounts.csv")
# 
# # Plot dispersions
# pdf("PN140_genotype-housing_dispersionPlot.pdf")
# plotDispEsts(dds3)
# dev.off()

```

# All Samples --------------------------------------------------------------------------------------
# All samples effect of genotype, removing housing and time effect (dds4)
```{r}
setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\all_samples\\genotype-time-housing\\")

# Create DESeq object
dds4 = DESeqDataSetFromMatrix(countData = counts, colData = coldata, design = ~housing + time + genotype)

# Set "reference" levels
dds4$housing <- relevel(dds4$housing, ref = "same")
dds4$genotype <- relevel(dds4$genotype, ref = "ctrl")
dds4$time <- relevel(dds4$time, ref = "PN60")

# Run DESeq on the DESeq object
dds4 = DESeq(dds4)
resallg <- results(dds4)
normed4 = counts(dds4, normalized=TRUE)

# # Check individual genes
# plotCounts(dds4,gene="Rnf122",intgroup="genotype")
# 
# # Write out results
# write.csv(as.data.frame(resallg), file="all_genotype-housing-time_results.csv")
# 
# # Create MA Plot, alpha is FDR cutoff, for this experiment it is "0.05"
# pdf("all_genotype-housing-time_MAPlot.pdf")
# DESeq2::plotMA(resallg, alpha = 0.05, ylim=c(-3,3), cex=.4)
# abline(h=c(-1,1), col="dodgerblue", lwd=2)
# dev.off()
# 
# # Write out normalized counts
# write.csv(normed4, file="all_normCounts.csv")
# 
# # Plot dispersions
# pdf("all_genotype-housing-time_dispersionPlot.pdf")
# plotDispEsts(dds4)
# dev.off()

```

# All samples effect of housing, removing genotype and time effect (dds5)
```{r}
setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\all_samples\\housing-genotype-time\\")

# Create DESeq object
dds5 = DESeqDataSetFromMatrix(countData = counts, colData = coldata, design = ~genotype + time + housing)

# Set "reference" levels
dds5$housing <- relevel(dds5$housing, ref = "same")
dds5$genotype <- relevel(dds5$genotype, ref = "ctrl")
dds5$time <- relevel(dds5$time, ref = "PN60")

# Run DESeq on the DESeq object
dds5 = DESeq(dds5)
resallh <- results(dds5)
normed5 = counts(dds5, normalized=TRUE)

# # Check individual genes
# plotCounts(dds5,gene="Rnf122",intgroup="genotype")
# 
# # Write out results
# write.csv(as.data.frame(resallh), file="all_housing-genotype-time_results.csv")
# 
# # Create MA Plot, alpha is FDR cutoff, for this experiment it is "0.05"
# pdf("all_housing-genotype-time_MAPlot.pdf")
# DESeq2::plotMA(resallh, alpha = 0.05, ylim=c(-3,3), cex=.4)
# abline(h=c(-1,1), col="dodgerblue", lwd=2)
# dev.off()
# 
# # Write out normalized counts
# write.csv(normed5, file="all_normCounts.csv")
# 
# # Plot dispersions
# pdf("all_housing-genotype-time_dispersionPlot.pdf")
# plotDispEsts(dds5)
# dev.off()

```

# PN140 SS animals only (dds10) (effect of genotype on ss housed animals)
```{r}
setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\PN140\\ss_genotype\\")

# Create DESeq object
dds10 = DESeqDataSetFromMatrix(countData = PN140SScounts, colData = PN140SScoldata, design = ~genotype)

# Set "reference" levels
# dds10$housing <- relevel(dds10$housing, ref = "same")
dds10$genotype <- relevel(dds10$genotype, ref = "ctrl")

# Run DESeq on the DESeq object
dds10 = DESeq(dds10)
res140SS <- results(dds10)
normed10 = counts(dds10, normalized=TRUE)

# # Check individual genes
# plotCounts(dds10,gene="Ccl5",intgroup="housing")

# Write out results
# write.csv(as.data.frame(res140SS), file="PN140SS_genotype_results.csv")
# 
# # Create MA Plot, alpha is FDR cutoff, for this experiment it is "0.05"
# pdf("PN140SS_genotype_MAPlot.pdf")
# DESeq2::plotMA(res140SS, alpha = 0.05, ylim=c(-3,3), cex=.4)
# abline(h=c(-1,1), col="dodgerblue", lwd=2)
# dev.off()
# 
# # Write out normalized counts
# write.csv(normed10, file="PN140SS_normCounts.csv")
# 
# # Plot dispersions
# pdf("PN140SS_genotype_dispersionPlot.pdf")
# plotDispEsts(dds10)
# dev.off()

normed10[rownames(normed10)=="Gpnmb",]
barplot(normed10[rownames(normed10)=="Gpnmb",])

```

# PN60 -------------------------------------------------------------------------------------------

# PN60 dnFGFR animals only (dds6) (effect of housing on dnFGFR animals)
```{r}
setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\PN60\\dn_housing\\")

# Create DESeq object
dds6 = DESeqDataSetFromMatrix(countData = PN60DNcounts, colData = PN60DNcoldata, design = ~housing)

# Set "reference" levels
dds6$housing <- relevel(dds6$housing, ref = "same")

# Run DESeq on the DESeq object
dds6 = DESeq(dds6)
res60DN <- results(dds6)
normed6 = counts(dds6, normalized=TRUE)

# # Check individual genes
# plotCounts(dds6,gene="Tnf",intgroup="housing")
# 
# # Write out results
# write.csv(as.data.frame(res60DN), file="PN60DN_housing_results.csv")
# 
# # Create MA Plot, alpha is FDR cutoff, for this experiment it is "0.05"
# pdf("PN60DN_housing_MAPlot.pdf")
# DESeq2::plotMA(res60DN, alpha = 0.05, ylim=c(-3,3), cex=.4)
# abline(h=c(-1,1), col="dodgerblue", lwd=2)
# dev.off()
# 
# # Write out normalized counts
# write.csv(normed6, file="PN60DN_normCounts.csv")
# 
# # Plot dispersions
# pdf("PN60DN_housing_dispersionPlot.pdf")
# plotDispEsts(dds6)
# dev.off()

```

# PN60 effect of housing, removing genotype effect (dds7)
```{r}
setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\PN60\\housing-gen\\")

# Create DESeq object
dds7 = DESeqDataSetFromMatrix(countData = PN60counts, colData = PN60coldata, design = ~genotype + housing)

# Set "reference" levels
dds7$housing <- relevel(dds7$housing, ref = "same")
dds7$genotype <- relevel(dds7$genotype, ref = "ctrl")

# Run DESeq on the DESeq object
dds7 = DESeq(dds7)
res60h <- results(dds7)
normed7 = counts(dds7, normalized=TRUE)

# # Check individual genes
# plotCounts(dds7,gene="Tnf",intgroup="housing")
# 
# # Write out results
# write.csv(as.data.frame(res60h), file="PN60_housing-gen_results.csv")
# 
# # Create MA Plot, alpha is FDR cutoff, for this experiment it is "0.05"
# pdf("PN60_housing-gen_MAPlot.pdf")
# DESeq2::plotMA(res60h, alpha = 0.05, ylim=c(-3,3), cex=.4)
# abline(h=c(-1,1), col="dodgerblue", lwd=2)
# dev.off()
# 
# # Write out normalized counts
# write.csv(normed7, file="PN60_normCounts.csv")
# 
# # Plot dispersions
# pdf("PN60_housing-gen_dispersionPlot.pdf")
# plotDispEsts(dds7)
# dev.off()

```

# PN60 effect of genotype, removing housing effect (dds8)
```{r}
setwd("C:\\GitHub\\dnFGFR\\deseq_outputs\\PN60\\genotype-housing\\")

# Create DESeq object
dds8 = DESeqDataSetFromMatrix(countData = PN60counts, colData = PN60coldata, design = ~housing + genotype)

# Set "reference" levels
dds8$housing <- relevel(dds8$housing, ref = "same")
dds8$genotype <- relevel(dds8$genotype, ref = "ctrl")

# Run DESeq on the DESeq object
dds8 = DESeq(dds8)
res60g <- results(dds8)
normed8 = counts(dds8, normalized=TRUE)

# # Check individual genes
# plotCounts(dds8,gene="Rnf122",intgroup="genotype")
# 
# # Write out results
# write.csv(as.data.frame(res60g), file="PN60_genotype-housing_results.csv")
# 
# # Create MA Plot, alpha is FDR cutoff, for this experiment it is "0.05"
# pdf("PN60_genotype-housing_MAPlot.pdf")
# DESeq2::plotMA(res60g, alpha = 0.05, ylim=c(-3,3), cex=.4)
# abline(h=c(-1,1), col="dodgerblue", lwd=2)
# dev.off()
# 
# # Write out normalized counts
# 
# write.csv(normed8, file="PN60_normCounts.csv")
# 
# # Plot dispersions
# pdf("PN60_genotype-housing_dispersionPlot.pdf")
# plotDispEsts(dds8)
# dev.off()

```









